url: https://www.nature.com/articles/s41551-025-01463-z
title: www.nature.com_articles_s41551-025-01463-z
authors: None
[Skip to main content](#content)

Thank you for visiting nature.com. You are using a browser version with limited support for CSS. To obtain
the best experience, we recommend you use a more up to date browser (or turn off compatibility mode in
Internet Explorer). In the meantime, to ensure continued support, we are displaying the site without styles
and JavaScript.

Advertisement

[![Advertisement](//pubads.g.doubleclick.net/gampad/ad?iu=/285/natbiomedeng.nature.com/article&sz=728x90&c=-1067444290&t=pos%3Dtop%26type%3Darticle%26artid%3Ds41551-025-01463-z%26doi%3D10.1038/s41551-025-01463-z%26techmeta%3D1,109,13,22,23,31,38,39,41,42,44,45,56,90%26subjmeta%3D1042,114,208,337,4041,631,639,705%26kwrd%3DComputational+biology+and+bioinformatics,Computational+science,CRISPR-Cas+systems)](//pubads.g.doubleclick.net/gampad/jump?iu=/285/natbiomedeng.nature.com/article&sz=728x90&c=-1067444290&t=pos%3Dtop%26type%3Darticle%26artid%3Ds41551-025-01463-z%26doi%3D10.1038/s41551-025-01463-z%26techmeta%3D1,109,13,22,23,31,38,39,41,42,44,45,56,90%26subjmeta%3D1042,114,208,337,4041,631,639,705%26kwrd%3DComputational+biology+and+bioinformatics,Computational+science,CRISPR-Cas+systems)

[![Nature Biomedical Engineering](https://media.springernature.com/full/nature-cms/uploads/product/natbiomedeng/header-8e114633cc891906a22898658743d641.svg)](/natbiomedeng)

* [View all journals](https://www.nature.com/siteindex)
* [Search](#search-menu)
* [Log in](https://idp.nature.com/auth/personal/springernature?redirect_uri=https://www.nature.com/articles/s41551-025-01463-z)

* [Explore content](#explore)
* [About the journal](#about-the-journal)
* [Publish with us](#publish-with-us)

* [Sign up for alerts](https://journal-alerts.springernature.com/subscribe?journal_id=41551)
* [RSS feed](https://www.nature.com/natbiomedeng.rss)

1. [nature](/)
2. [nature biomedical engineering](/natbiomedeng)
3. [articles](/natbiomedeng/articles?type=article)
4. article

CRISPR-GPT for agentic automation of gene-editing experiments

[Download PDF](/articles/s41551-025-01463-z.pdf)

[Download PDF](/articles/s41551-025-01463-z.pdf)

* Article
* [Open access](https://www.springernature.com/gp/open-science/about/the-fundamentals-of-open-access-and-open-research)
* Published: 30 July 2025

# CRISPR-GPT for agentic automation of gene-editing experiments

* [Yuanhao Qu](#auth-Yuanhao-Qu-Aff1)
  [ORCID: orcid.org/0000-0002-7574-259X](https://orcid.org/0000-0002-7574-259X)[1](#Aff1)[na1](#na1),
* [Kaixuan Huang](#auth-Kaixuan-Huang-Aff2)[2](#Aff2)[na1](#na1),
* [Ming Yin](#auth-Ming-Yin-Aff2)[2](#Aff2),
* [Kanghong Zhan](#auth-Kanghong-Zhan-Aff3)[3](#Aff3),
* [Dyllan Liu](#auth-Dyllan-Liu-Aff4)[4](#Aff4),
* [Di Yin](#auth-Di-Yin-Aff1)[1](#Aff1),
* [Henry C. Cousins](#auth-Henry_C_-Cousins-Aff5-Aff6)[5](#Aff5),[6](#Aff6),
* [William A. Johnson](#auth-William_A_-Johnson-Aff1)[1](#Aff1),
* [Xiaotong Wang](#auth-Xiaotong-Wang-Aff1)[1](#Aff1),
* [Mihir Shah](#auth-Mihir-Shah-Aff5)
  [ORCID: orcid.org/0000-0003-0854-2384](https://orcid.org/0000-0003-0854-2384)[5](#Aff5),
* [Russ B. Altman](#auth-Russ_B_-Altman-Aff4-Aff7)
  [ORCID: orcid.org/0000-0003-3859-2905](https://orcid.org/0000-0003-3859-2905)[4](#Aff4),[7](#Aff7),
* [Denny Zhou](#auth-Denny-Zhou-Aff8)[8](#Aff8),
* [Mengdi Wang](#auth-Mengdi-Wang-Aff2)
  [ORCID: orcid.org/0000-0002-2101-9507](https://orcid.org/0000-0002-2101-9507)[2](#Aff2) &
* …
* [Le Cong](#auth-Le-Cong-Aff1)
  [ORCID: orcid.org/0000-0003-4725-8714](https://orcid.org/0000-0003-4725-8714)[1](#Aff1)

Show authors

[*Nature Biomedical Engineering*](/natbiomedeng)
(2025)[Cite this article](#citeas)

### Subjects

* [Computational biology and bioinformatics](/subjects/computational-biology-and-bioinformatics)
* [Computational science](/subjects/computational-science)
* [CRISPR-Cas systems](/subjects/crispr-cas-systems)

## Abstract

Performing effective gene-editing experiments requires a deep understanding of both the CRISPR technology and the biological system involved. Meanwhile, despite their versatility and promise, large language models (LLMs) often lack domain-specific knowledge and struggle to accurately solve biological design problems. We present CRISPR-GPT, an LLM agent system to automate and enhance CRISPR-based gene-editing design and data analysis. CRISPR-GPT leverages the reasoning capabilities of LLMs for complex task decomposition, decision-making and interactive human–artificial intelligence (AI) collaboration. This system incorporates domain expertise, retrieval techniques, external tools and a specialized LLM fine tuned with open-forum discussions among scientists. CRISPR-GPT assists users in selecting CRISPR systems, experiment planning, designing guide RNAs, choosing delivery methods, drafting protocols, designing assays and analysing data. We showcase the potential of CRISPR-GPT by knocking out four genes with CRISPR-Cas12a in a human lung adenocarcinoma cell line and epigenetically activating two genes using CRISPR-dCas9 in a human melanoma cell line. CRISPR-GPT enables fully AI-guided gene-editing experiment design and analysis across different modalities, validating its effectiveness as an AI co-pilot in genome engineering.

### Similar content being viewed by others

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41586-025-09298-z/MediaObjects/41586_2025_9298_Fig1_HTML.png)
### [Design of highly functional genome editors by modelling CRISPR–Cas sequences](https://www.nature.com/articles/s41586-025-09298-z?fromPaywallRec=false)

Article
Open access
30 July 2025

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41580-023-00697-6/MediaObjects/41580_2023_697_Fig1_HTML.png)
### [CRISPR technologies for genome, epigenome and transcriptome editing](https://www.nature.com/articles/s41580-023-00697-6?fromPaywallRec=false)

Article
02 February 2024

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41579-022-00739-4/MediaObjects/41579_2022_739_Fig1_HTML.png)
### [Structural biology of CRISPR–Cas immunity and genome editing enzymes](https://www.nature.com/articles/s41579-022-00739-4?fromPaywallRec=false)

Article
13 May 2022

## Main

Large language models (LLMs) have demonstrated exceptional capabilities in language skills and encapsulate a substantial amount of world knowledge[1](#ref-CR1 "Chowdhery, A. et al. PaLM: scaling language modeling with pathways. J. Mach. Learn. Res. 24, 240 (2023)."),[2](#ref-CR2 "Hoffmann, J. et al. An empirical analysis of compute-optimal large language model training. In Advances in Neural Information Processing Systems (eds Koyejo, S. et al.) Vol. 35, 30016–30030 (Curran Associates, Inc., 2022)."),[3](#ref-CR3 "OpenAI et al. GPT-4 technical report. Preprint at
                https://doi.org/10.48550/arXiv.2303.08774

               (2024)."),[4](#ref-CR4 "The Claude 3 Model Family: Opus, Sonnet, Haiku (Anthropic, 2024)."),[5](/articles/s41551-025-01463-z#ref-CR5 "Anil, R. et al. PaLM 2 technical report. Preprint at
                https://doi.org/10.48550/arXiv.2305.10403

               (2023)."). Recent research has also enhanced LLMs with external tools, improving their problem-solving abilities and efficiencies[6](#ref-CR6 "Yao, S. et al. ReAct: synergizing reasoning and acting in language models. In Proc. 11th International Conference on Learning Representations (OpenReview, 2023)."),[7](#ref-CR7 "Liu, R. et al. Mind’s eye: grounded language model reasoning through simulation. In Proc. 11th International Conference on Learning Representations (OpenReview, 2023)."),[8](/articles/s41551-025-01463-z#ref-CR8 "Schick, T. et al. Toolformer: language models can teach themselves to use tools. In Advances in Neural Information Processing Systems (eds Oh, A. et al.) Vol. 36, 68539–68551 (Curran Associates, Inc., 2023)."). Moreover, LLMs have also demonstrated potential as tool makers[9](/articles/s41551-025-01463-z#ref-CR9 "Cai, T., Wang, X., Ma, T., Chen, X. & Zhou, D. Large language models as tool makers. In Proc. 12th International Conference on Learning Representations (OpenReview, 2024).") and black-box optimizers[10](/articles/s41551-025-01463-z#ref-CR10 "Yang, C. et al. Large language models as optimizers. In Proc. 12th International Conference on Learning Representations (OpenReview, 2024)."). To this end, researchers have explored LLM-based specialized models for various scientific domains[11](/articles/s41551-025-01463-z#ref-CR11 "Li, T. et al. CancerGPT for few shot drug pair synergy prediction using large pretrained language models. npj Digit. Med. 7, 40 (2024)."),[12](/articles/s41551-025-01463-z#ref-CR12 "Wu, S. et al. BloombergGPT: a large language model for finance. Preprint at
                https://doi.org/10.48550/arXiv.2303.17564

               (2023)."), particularly for mathematics and chemistry tasks. ChemCrow[13](/articles/s41551-025-01463-z#ref-CR13 "M. Bran, A. et al. Augmenting large language models with chemistry tools. Nat. Mach. Intell. 6, 525–535 (2024).") uses tool-augmented LLM for solving a range of chemistry-related tasks such as paracetamol synthesis, whereas Co-scientist[14](/articles/s41551-025-01463-z#ref-CR14 "Boiko, D. A., MacKnight, R., Kline, B. & Gomes, G. Autonomous chemical research with large language models. Nature 624, 570–578 (2023).") integrates automated experimentation, achieving successful optimization of palladium-catalysed cross-coupling reaction. LLMs have also shown initial promise in generating biological protocols, as demonstrated by studies like BioPlanner[15](/articles/s41551-025-01463-z#ref-CR15 "O’Donoghue, O. et al. BioPlanner: automatic evaluation of LLMs on protocol planning in biology. In Proceedings of the 2023 Conference on Empirical Methods in Natural Language Processing (eds Bouamor, H., Pino, J. & Bali, K.) 2676–2694 (Association for Computational Linguistics, 2023)."). While recent advancements, such as OpenAI’s o1 preview, have improved reasoning abilities in areas such as mathematics and coding, progress in biological tasks remains comparatively limited. This limitation stems from general-purpose LLMs’ lack of in-depth understanding of biology, compounded by the unique challenges of biological experiments, including the variability of living systems, the noisy nature of biological data and the highly specialized, less transferable nature of biological skills and tools.

Gene editing has transformed biological research and medicine, allowing for precise DNA modifications for both therapeutic and experimental applications. CRISPR-Cas, the most well-known gene-editing technology, originated from bacterial immune systems[16](#ref-CR16 "Jinek, M. et al. A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity. Science 337, 816–821 (2012)."),[17](#ref-CR17 "Cong, L. et al. Multiplex genome engineering using CRISPR/Cas systems. Science 339, 819–823 (2013)."),[18](#ref-CR18 "Mali, P. et al. RNA-guided human genome engineering via Cas9. Science 339, 823–826 (2013)."),[19](#ref-CR19 "Jinek, M. et al. RNA-programmed genome editing in human cells. eLife 2, e00471 (2013)."),[20](#ref-CR20 "Doudna, J. A. & Charpentier, E. The new frontier of genome engineering with CRISPR-Cas9. Science 346, 1258096 (2014)."),[21](#ref-CR21 "Mali, P., Esvelt, K. M. & Church, G. M. Cas9 as a versatile tool for engineering biology. Nat. Methods 10, 957–963 (2013)."),[22](#ref-CR22 "Hsu, P. D., Lander, E. S. & Zhang, F. Development and applications of CRISPR-Cas9 for genome engineering. Cell 157, 1262–1278 (2014)."),[23](#ref-CR23 "Sander, J. D. & Joung, J. K. CRISPR-Cas systems for editing, regulating and targeting genomes. Nat. Biotechnol. 32, 347–355 (2014)."),[24](/articles/s41551-025-01463-z#ref-CR24 "Shalem, O., Sanjana, N. E. & Zhang, F. High-throughput functional genomics using CRISPR–Cas9. Nat. Rev. Genet. 16, 299–311 (2015)."). Its development has led to advanced techniques like CRISPR activation and interference (CRISPRa/i)[25](#ref-CR25 "Qi, L. S. et al. Repurposing CRISPR as an RNA-guided platform for sequence-specific control of gene expression. Cell 152, 1173–1183 (2013)."),[26](#ref-CR26 "Gilbert, L. A. et al. CRISPR-mediated modular RNA-guided regulation of transcription in eukaryotes. Cell 154, 442–451 (2013)."),[27](#ref-CR27 "Konermann, S. et al. Optical control of mammalian endogenous transcription and epigenetic states. Nature 500, 472–476 (2013)."),[28](#ref-CR28 "Maeder, M. L. et al. CRISPR RNA-guided activation of endogenous human genes. Nat. Methods 10, 977–979 (2013)."),[29](/articles/s41551-025-01463-z#ref-CR29 "Perez-Pinera, P. et al. RNA-guided gene activation by CRISPR-Cas9-based transcription factors. Nat. Methods 10, 973–976 (2013)."), base editing[30](/articles/s41551-025-01463-z#ref-CR30 "Anzalone, A. V. et al. Search-and-replace genome editing without double-strand breaks or donor DNA. Nature 576, 149–157 (2019)."),[31](/articles/s41551-025-01463-z#ref-CR31 "Gaudelli, N. M. et al. Programmable base editing of A•T to G•C in genomic DNA without DNA cleavage. Nature 551, 464–471 (2017).") and prime editing[32](/articles/s41551-025-01463-z#ref-CR32 "Anzalone, A. V., Koblan, L. W. & Liu, D. R. Genome editing with CRISPR–Cas nucleases, base editors, transposases and prime editors. Nat. Biotechnol. 38, 824–844 (2020)."),[33](/articles/s41551-025-01463-z#ref-CR33 "Pickar-Oliver, A. & Gersbach, C. A. The next generation of CRISPR–Cas technologies and applications. Nat. Rev. Mol. Cell Biol. 20, 490–507 (2019)."), creating a powerful toolkit for genetic modification and epigenetic modulation. In basic biomedical research, CRISPR gene-editing has become one of the most frequently used laboratory techniques: at the largest non-profit plasmid DNA repository, Addgene, 8 of the 15 top requested plasmids worldwide were for CRISPR gene-editing[34](/articles/s41551-025-01463-z#ref-CR34 "Tsang, J. 15 Years of Addgene: The Top 15 Plasmids. Addgene Blog
                https://blog.addgene.org/15-years-of-addgene-the-top-15-plasmids

               (2019)."). On the application side, CRISPR has produced the first permanent cure for sickle cell disease (SCD)[35](/articles/s41551-025-01463-z#ref-CR35 "Office of the Commissioner. FDA approves first gene therapies to treat patients with sickle cell disease. FDA News Release
                https://www.fda.gov/news-events/press-announcements/fda-approves-first-gene-therapies-treat-patients-sickle-cell-disease

               (2024).") and β-thalassaemia[36](/articles/s41551-025-01463-z#ref-CR36 "Office of the Commissioner. FDA roundup: January 16, 2024. FDA News Release
                https://www.fda.gov/news-events/press-announcements/fda-roundup-january-16-2024

               (2024)."), as well as facilitating plant engineering for sustainable agriculture[20](/articles/s41551-025-01463-z#ref-CR20 "Doudna, J. A. & Charpentier, E. The new frontier of genome engineering with CRISPR-Cas9. Science 346, 1258096 (2014)."). As one of the most powerful biotechnologies, numerous software and protocols exist for specific gene-editing tasks. Despite these resources, an end-to-end solution—from CRISPR-Cas system selection, guide (g)RNA design, off-target evaluation, to delivery and data analysis—remains complex, particularly for newcomers. AI-assisted tools can simplify gene-editing experiment design and data analysis, making the technology more accessible and accelerating scientific and therapeutic discoveries.

We introduce CRISPR-GPT, a solution that combines the strengths of LLMs with domain-specific knowledge, chain-of-thought reasoning, instruction fine-tuning, retrieval techniques and tools. CRISPR-GPT is centred around LLM-powered planning and execution agents (Fig. [1](/articles/s41551-025-01463-z#Fig1)). This system leverages the reasoning abilities of general-purpose LLMs and multi-agent collaboration for task decomposition, constructing state machines and automated decision-making (Fig. [2a](/articles/s41551-025-01463-z#Fig2)). It draws upon expert knowledge from leading practitioners and peer-reviewed published literature in gene editing for retrieval-augmented generation (RAG)[13](/articles/s41551-025-01463-z#ref-CR13 "M. Bran, A. et al. Augmenting large language models with chemistry tools. Nat. Mach. Intell. 6, 525–535 (2024).").

**Fig. 1: Overview of CRISPR-GPT.**

[![figure 1](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig1_HTML.png)](/articles/s41551-025-01463-z/figures/1)

CRISPR-GPT is an LLM-powered multi-agent system designed to provide AI copiloting for human researchers in gene editing. It supports four primary gene-editing modalities: knockout, base editing, prime editing and epigenetic editing (CRISPRa/i). The system offers three user interaction modes: Meta mode (step-by-step guidance on predefined tasks), Auto mode (customized guidance based on user requests) and Q&A mode (real-time answers to ad hoc questions), to streamline experiment design and planning. CRISPR-GPT consists of four core components: the User proxy, LLM planner, Task executor and Tool provider. Together, these components are equipped with a comprehensive suite of tools and decision-support capabilities to facilitate the design, planning and analysis of gene-editing workflows. To evaluate CRISPR-GPT’s performance, we developed the Gene-editing bench, a framework of 288 test cases covering tasks such as experimental planning, sgRNA design, delivery method selection and more. Figure was originally created with [BioRender.com/tb8sq6f](https://BioRender.com/tb8sq6f).

[Full size image](/articles/s41551-025-01463-z/figures/1)

**Fig. 2: CRISPR-GPT adopts a compositional, multi-agent architecture to enable human–AI collaboration and automated experimental designs.**

[![figure 2](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig2_HTML.png)](/articles/s41551-025-01463-z/figures/2)

**a**, The backbone of CRISPR-GPT involves multi-agent collaboration between four core components: (1) The LLM Planner agent is responsible for configuring tasks on the basis of the user’s needs. It automatically performs task decomposition on the basis of the user’s request, the descriptions of the currently supported tasks and internal knowledge. The state machines of the selected tasks are chained together to fulfill the user’s request. (2) The Task executor agent implements the chain of state machines from the Planner agent and is responsible for providing instructions and feedback, receiving input from the User-proxy agent and calling external tools. State machines are central to the Task executor, where each state is responsible for one round of interaction with the user. The instruction is provided to the user first with sufficient information for the current decision-making step and the required inputs. After receiving the response from the user, it provides output and feedback, where Tool providers are potentially called during the execution of the state. Afterwards, the state machine transits to the next state. (3) The LLM User-proxy agent is responsible for interacting with the Task executor on behalf of the user, where the user can monitor the process and provide corrections to the User-proxy agent if the generated content needs modification or improvement. It generates responses to every step of the state machine on behalf of the user. (4) Tool providers support diverse external tools and connect to search engines or databases via API calls. Part of the panel was created with [BioRender.com/svkmgjk](https://BioRender.com/svkmgjk). **b**, Breakdown of individual tasks in a typical CRISPR-GPT workflow for gene-editing experiments.

[Full size image](/articles/s41551-025-01463-z/figures/2)
## Results

### Building AI co-pilot harnessing LLM’s reasoning ability

CRISPR-GPT supports four major gene-editing modalities and 22 gene-editing experiment tasks (Fig. [1](/articles/s41551-025-01463-z#Fig1) and Supplementary Table [1](/articles/s41551-025-01463-z#MOESM1)). It offers tunable levels of automation via three modes: Meta, Auto and Q&A. They are designed to accommodate users ranging from novice PhD-level scientists fresh to gene editing, to domain experts looking for more efficient, automated solutions for selected tasks (Fig. [1](/articles/s41551-025-01463-z#Fig1)). The ‘Meta mode’ is designed for beginner researchers, guiding them through a sequence of essential tasks from selection of CRISPR systems, delivery methods, to designing gRNA, assessing off-target efficiency, generating experiment protocols and data analysis. Throughout this decision-making process, CRISPR-GPT interacts with users at every step, provides instructions and seeks clarifications when needed. The ‘Auto mode’ caters to advanced researchers and does not adhere to a predefined task order. Users submit a freestyle request, and the LLM Planner decomposes this into tasks, manages their interdependence, builds a customized workflow and executes them automatically. It fills in missing information on the basis of the initial inputs and explains its decisions and thought process, allowing users to monitor and adjust the process. The ‘Q&A mode’ supports users with on-demand scientific inquiries about gene editing.

To assess the AI agent’s capabilities to perform gene-editing research, we compiled an evaluation test set, Gene-editing bench, from both public sources and human experts (details in Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). This test set covers a variety of gene-editing tasks (Fig. [1](/articles/s41551-025-01463-z#Fig1)). By using the test set, we performed extensive evaluation of CRISPR-GPT’s capabilities in major gene-editing research tasks, such as experiment planning, delivery selection, single guide (sg)RNA design and experiment troubleshooting. In addition, we invited human experts to perform a thorough user experience evaluation of CRISPR-GPT and collected valuable human feedback.

Further, we implement CRISPR-GPT in real-world wet labs. Using CRISPR-GPT as an AI co-pilot, we demonstrate a fully AI-guided knockout (KO) of four genes: *TGFβR1*, *SNAI1*, *BAX* and *BCL2L1*, using CRISPR-Cas12a in human lung adenocarcinoma cell line, as well as AI-guided CRISPR-dCas9 epigenetic activation of two genes: *NCR3LG1* and *CEACAM1*, in a human melanoma model. All these wet-lab experiments were carried out by junior researchers not familiar with gene editing. They both succeeded on the first attempt, confirmed by not only editing efficiencies, but also biologically relevant phenotypes and protein-level validation, highlighting the potential of LLM-guided biological research.

CRISPR-GPT is a multi-agent, compositional system involving a team of LLM-based agents, including an LLM Planner agent, a User-proxy agent, Task executor agents and Tool provider agents (Fig. [2a](/articles/s41551-025-01463-z#Fig2)). These components are powered by LLMs to interact with one another as well as the human user. We also refer to the full system as an ‘agent’ to encapsulate the overall functionalities.

To automate biological experiment design and analysis, we view the overall problem as sequential decision-making. This perspective frames the interaction between the user and the automated system as a series of decision-making steps, each essential for progressing towards the ultimate goal. Take the Auto mode for example. A user can initiate the process with a meta-request, for example, “I want to knock out the human TGFβR1 gene in A549 lung cancer cells”. In response, the agent’s LLM Planner will analyse the user’s request, drawing on its extensive internal knowledge base via retrieval techniques. Leveraging the reasoning abilities of the base LLM, the Planner generates a chain-of-thought[37](/articles/s41551-025-01463-z#ref-CR37 "Wei, J. et al. Chain-of-thought prompting elicits reasoning in large language models. In Advances in Neural Information Processing Systems (eds Koyejo, S. et al.) Vol. 35, 24824–24837 (Curran Associates, Inc., 2022)."),[38](/articles/s41551-025-01463-z#ref-CR38 "Zhou, D. et al. Least-to-most prompting enables complex reasoning in large language models. In Proc. 11th International Conference on Learning Representations (OpenReview, 2023).") reasoning path and chooses an optimal action from a set of plausible ones while following expert-written guidelines. Consequently, the Planner breaks down the user’s request into a sequence of discrete tasks, for example, ‘CRISPR-Cas system selection’ and ‘gRNA design for knockout’, while managing interdependencies among these tasks. Each individual task is solved by an LLM-powered state machine, via the Task executor, entailing a sequence of states to progress towards the specific goal. After the meta-task decomposition, the Task executor will chain the state machines of the corresponding tasks together into a larger state machine and begin the execution process, systematically addressing each task in sequence to ensure that the experiment’s objectives are met efficiently and effectively (Fig. [2a](/articles/s41551-025-01463-z#Fig2)).

The User-proxy agent is responsible for guiding the user throughout the decision-making process via multiple rounds of textual interactions (typical user interactions required by each task detailed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). At each decision point, the internal state machine presents a ‘state variable’ to the User-proxy agent, which includes the current task instructions, and specifies any necessary input from the user to proceed. The User-proxy agent then interprets this state given the user interactions and makes informed decisions as input to the Task executor on behalf of the user. Subsequently, the User-proxy agent receives feedback from the Task executor, including the task results and the reasoning process that led to those outcomes. Concurrently, the User-proxy agent continues to interact with the user and provides them with instructions, continuously integrating their feedback to ensure alignment with the user’s objectives (detailed in [Methods](/articles/s41551-025-01463-z#Sec13); Fig. [2a](/articles/s41551-025-01463-z#Fig2) and Supplementary Fig. [1](/articles/s41551-025-01463-z#MOESM1)).

To enhance the LLM with domain knowledge, we enable the CRISPR agent to retrieve and synthesize information from published protocols, peer-reviewed research papers and expert-written guidelines, and to utilize external tools and conduct web searches via Tool provider agents (Fig. [2a](/articles/s41551-025-01463-z#Fig2)).

For an end-to-end gene-editing workflow, CRISPR-GPT typically constructs a chain of tasks that includes selecting the appropriate CRISPR system, recommending delivery methods, designing gRNAs, predicting off-target effects, selecting experimental protocols, planning validation assays and performing data analysis (Fig. [2b](/articles/s41551-025-01463-z#Fig2)). The system’s modular architecture facilitates easy integration of additional functionalities and tools. CRISPR-GPT serves as a prototype LLM-powered AI co-pilot for scientific research, with potential applications extending beyond gene editing.

### CRISPR-GPT agents automate gene-editing research tasks

CRISPR-GPT is able to automate gene-editing research via several key functionalities. For each functionality we discuss the agentic implementation and evaluation results.

#### Experiment planning

The Task planner agent is charged with directing the entire workflow and breaking down the user’s meta-request into a task chain (Fig. [2b](/articles/s41551-025-01463-z#Fig2)). While the Planner selects and follows a predefined workflow in the Meta mode, it is able to take in freestyle user requests and auto-build a customized workflow in the Auto mode. For example, a user may only need part of the pre-designed workflow including CRISPR-Cas system selection, delivery method selection, gRNA design and experimental protocol selection before the experiment. Then the Task planner agent extracts the right information from the user request and assembles a customized workflow to suit user needs (Fig. [3a](/articles/s41551-025-01463-z#Fig3)). To evaluate CRISPR-GPT’s ability to correctly layout gene-editing tasks and manage intertask dependence, we compiled a planning test set, as part of the Gene-editing bench, with user requests and golden answers curated by human experts. Using this test set, we evaluated CRISPR-GPT in comparison with prompted general LLMs, showing that CRISPR-GPT outperforms general LLMs in planning gene-editing tasks (Fig. [3b](/articles/s41551-025-01463-z#Fig3)). The CRISPR-GPT agent driven by GPT-4o scored over 0.99 in accuracy, precision, recall and F1 score, and had <0.05 normalized Levenshtein distance between agent-generated plans and golden answers (Fig. [3b](/articles/s41551-025-01463-z#Fig3)). For extensive description of the test set and evaluation, please see Supplementary Note [C1](/articles/s41551-025-01463-z#MOESM1).

**Fig. 3: Task decomposition and experiment planning in CRISPR-GPT Auto mode with performance evaluation.**

[![figure 3](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig3_HTML.png)](/articles/s41551-025-01463-z/figures/3)

**a**, The LLM Planner agent automatically breaks down the user’s meta-request to a sequence of tasks. Then it assembles a customized workflow of the chained tasks to meet the user’s needs. Part of the panel was created with [BioRender.com/qy4v](https://BioRender.com/qy4v). **b**, Evaluation of the LLM Planner using a gene-editing planning test set. For each test case, we generate three independent answers from each model and report the average scores (Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). Data shown are the mean ± s.d. (*n* = 3 per group).

[Full size image](/articles/s41551-025-01463-z/figures/3)
#### Delivery method selection

We present and evaluate the delivery agent of CRISPR-GPT (Fig. [4a,b](/articles/s41551-025-01463-z#Fig4)). Delivery is a critical step for all gene-editing experiments. CRISPR-GPT equips LLM with expert-tailored instructions and external tools to choose delivery methods. Specifically, the agent first tries to understand the biological system that the user is planning to edit. It extracts keywords for the target cell/tissues/organisms, performs Google search and summarizes the results. Then, given its own knowledge and search results, CRISPR-GPT matches the user case with a major biological category (cell lines, primary cells, in vivo and so on) which reduces the possible options to a focused set of candidate methods. Next, CRISPR-GPT performs literature search with user and method-specific keywords, and ranks the candidate methods on the basis of citation metrics to suggest a primary and a secondary delivery method (Fig. [4a](/articles/s41551-025-01463-z#Fig4)). To evaluate the performance of this module, we compiled test cases including 50 biological systems as part of the Gene-editing bench. For each case, we invited three human experts to score potential delivery options and utilized those as ground truth. We then evaluated the output of CRISPR-GPT and baseline models by comparing to the pre-compiled ground-truth score sheet. We found that CRISPR-GPT outperforms the baseline GPT-4 and GPT-3-turbo models (Fig. [4b](/articles/s41551-025-01463-z#Fig4)). The agent has a substantial edge on difficult tasks such as those involving hard-to-transfect cell lines and primary cell types. We also noticed that including an additional literature search step improves the agent’s performance only moderately. More details about the delivery selection evaluation can be found in Supplementary Note [C2](/articles/s41551-025-01463-z#MOESM1).

**Fig. 4: CRISPR-GPT automates gene-editing research and experiment tasks.**

[![figure 4](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig4_HTML.png)](/articles/s41551-025-01463-z/figures/4)

**a**, Design of delivery method selection agent in CRISPR-GPT, showing the workflow, example request and a series of agent thoughts–actions to identify most suitable delivery methods for the user’s needs. **b**, Evaluation results of delivery method selection using CRISPR-GPT and baseline models. Data shown are the mean ± s.d. (*n* = 3 per group). **c**, Design of gRNA design agent in CRISPR-GPT, showing the workflow, example request and a series of agent thoughts–actions to select top-ranked gRNA customized to user’s request. **d**, Evaluation results of gRNA design using CRISPR-GPT and baseline models. Models were prompted to generate functions and associated parameters to design gRNAs requested by the user. Data shown are the mean ± s.d. (*n* = 3 per group). **e**, Design of Q&A mode in CRISPR-GPT, showing the workflow, example request and a series of agent thoughts–actions to answer gene-editing questions. **f**, Evaluation of CRISPR-GPT and baseline models for answering gene-editing research questions. Models were prompted to generate answers, which were anonymized, evaluated by three human experts in a fully blind setup. Scores range from 1 (lowest) to 5 (highest). All scores above were from three independent trials (details on evaluation in Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). Data shown are the mean ± s.d. (*n* = 3 per group).

[Full size image](/articles/s41551-025-01463-z/figures/4)
#### gRNA design

Good gRNA design is crucial for the success of CRISPR experiments. Various gRNA design tools and software, such as CRISPick[39](#ref-CR39 "Kim, H. K. et al. Deep learning improves prediction of CRISPR–Cpf1 guide RNA activity. Nat. Biotechnol. 36, 239–241 (2018)."),[40](#ref-CR40 "DeWeirdt, P. C. et al. Optimization of AsCas12a for combinatorial genetic screens in human cells. Nat. Biotechnol. 39, 94–104 (2021)."),[41](#ref-CR41 "Doench, J. G. et al. Optimized sgRNA design to maximize activity and minimize off-target effects of CRISPR-Cas9. Nat. Biotechnol. 34, 184–191 (2016)."),[42](/articles/s41551-025-01463-z#ref-CR42 "Sanson, K. R. et al. Optimized libraries for CRISPR-Cas9 genetic screens with multiple modalities. Nat. Commun. 9, 5416 (2018).") and ChopChop[43](/articles/s41551-025-01463-z#ref-CR43 "Labun, K. et al. CHOPCHOP v3: expanding the CRISPR web toolbox beyond genome editing. Nucleic Acids Res. 47, W171–W174 (2019)."), are available. However, we believe there are two key challenges in general usage: (1) choosing a trustworthy source and (2) difficulty in quickly identifying gRNAs that suit specific user requirements or experiment contexts, often requiring lengthy sorting, ranking or literature review. To address these issues, we utilized pre-designed gRNA tables from CRISPick, a reputable and widely used tool. We leverage the reasoning capabilities of LLMs to accurately identify regions of interest and quickly extract relevant gRNAs. This approach is similar to the recently proposed ‘chain-of-tables’ methodology[44](/articles/s41551-025-01463-z#ref-CR44 "Wang, Z. et al. Chain-of-table: evolving tables in the reasoning chain for table understanding. In Proc. 12th International Conference on Learning Representations (OpenReview, 2024).") (Fig. [4c](/articles/s41551-025-01463-z#Fig4), Extended Data Fig. [1a](/articles/s41551-025-01463-z#Fig7), and Supplementary Demo Videos [1](/articles/s41551-025-01463-z#MOESM5) and [2](/articles/s41551-025-01463-z#MOESM6)). To evaluate the ability of CRISPR-GPT to correctly retrieve gRNAs, we compiled a gRNA design test set with ground truth from human experts (detailed in Supplementary Note [C3](/articles/s41551-025-01463-z#MOESM1)). CRISPR-GPT agent outperforms the baseline LLMs in accurately selecting gRNA design actions and configuring the arguments (Fig. [4d](/articles/s41551-025-01463-z#Fig4)).

Further, we picked a real-world test case from a cancer biology study, in which many highly ranked gRNA designs did not generate biological phenotypes, even when their editing efficiencies were high[45](/articles/s41551-025-01463-z#ref-CR45 "Shi, J. et al. Discovery of cancer drug targets by CRISPR-Cas9 screening of protein domains. Nat. Biotechnol. 33, 661–667 (2015)."). Instead, the authors of the study had to design gRNAs manually against exons encoding important functional domains within a gene, and exon-selected gRNAs induced expected cancer-killing effects. We tested CRISPR-GPT for designing gRNAs targeting BRD4 gene from this study, and compared results with those generated by CRISPick and CHOPCHOP (Extended Data Fig. [1](/articles/s41551-025-01463-z#Fig7)). CRISPR-GPT was uniquely able to select the key exons, Exon3–4, within BRD4. In contrast, gRNAs designed by CRISPick or CHOPCHOP would be likely ineffective, as 7 out of 8 gRNAs mapped to non-essential exons (Extended Data Fig. [1](/articles/s41551-025-01463-z#Fig7)). Taken together, our results support the benefit and validity of this module.

#### Other functions and tools

CRISPR-GPT provides specific suggestions on the choice of the CRISPR system, experimental and validation protocol by leveraging LLM’s reasoning ability and retrieving information from an expert-reviewed knowledge base[46](#ref-CR46 "Giuliano, C. J., Lin, A., Girish, V. & Sheltzer, J. M. Generating single cell-derived knockout clones in mammalian cells with CRISPR/Cas9. Curr. Protoc. Mol. Biol. 128, e100 (2019)."),[47](#ref-CR47 "Du, D. & Qi, L. S. CRISPR technology for genome activation and repression in mammalian cells. Cold Spring Harb. Protoc. 2016, pdb.prot090175 (2016)."),[48](#ref-CR48 "Huang, T. P., Newby, G. A. & Liu, D. R. Precision genome editing using cytosine and adenine base editors in mammalian cells. Nat. Protoc. 16, 1089–1128 (2021)."),[49](#ref-CR49 "Hanna, R. E. et al. Massively parallel assessment of human variants with base editor screens. Cell 184, 1064–1080.e20 (2021)."),[50](#ref-CR50 "Doman, J. L., Sousa, A. A., Randolph, P. B., Chen, P. J. & Liu, D. R. Designing and executing prime editing experiments in mammalian cells. Nat. Protoc. 17, 2431–2468 (2022)."),[51](/articles/s41551-025-01463-z#ref-CR51 "McGee, A. V. et al. Modular vector assembly enables rapid assessment of emerging CRISPR technologies. Cell Genomics 4, 100519 (2024)."). It also offers automated gRNA off-target prediction, primer design for validation experiments, and data analysis. In particular, the agent provides fully automated solutions to run external software, such as Primer3 (ref. [52](/articles/s41551-025-01463-z#ref-CR52 "Untergasser, A. et al. Primer3—new capabilities and interfaces. Nucleic Acids Res. 40, e115 (2012).")), CRISPRitz[53](#ref-CR53 "Cancellieri, S., Canver, M. C., Bombieri, N., Giugno, R. & Pinello, L. CRISPRitz: rapid, high-throughput and variant-aware in silico off-target site identification for CRISPR genome editing. Bioinformatics 36, 2001–2008 (2020)."),[54](#ref-CR54 "Hwang, G.-H. et al. PE-Designer and PE-Analyzer: web-based design and analysis tools for CRISPR prime editing. Nucleic Acids Res. 49, W499–W504 (2021)."),[55](#ref-CR55 "Chow, R. D., Chen, J. S., Shen, J. & Chen, S. A web tool for the design of prime-editing guide RNAs. Nat. Biomed. Eng. 5, 190–194 (2021)."),[56](/articles/s41551-025-01463-z#ref-CR56 "Morris, J. A., Rahman, J. A., Guo, X. & Sanjana, N. E. Automated design of CRISPR prime editors for 56,000 human pathogenic variants. iScience 24, 103380 (2021).") and CRISPResso2 (ref. [57](/articles/s41551-025-01463-z#ref-CR57 "Clement, K. et al. CRISPResso2 provides accurate and rapid genome editing sequence analysis. Nat. Biotechnol. 37, 224–226 (2019).")) (Supplementary Table [1](/articles/s41551-025-01463-z#MOESM1)). We focused on implementing these tools as they are considered gold standard in respective tasks and have been extensively validated in previous work.

### Problem solving via fine-tuning LLM on scientific discussion

General-purpose LLMs may possess broad knowledge but often lack the deep understanding of science needed to solve research problems. To enhance the CRISPR-GPT agent’s capacity in answering advanced research questions, we build a Q&A mode that synthesizes information from multiple resources, including published literature, established protocols and discussions between human scientists, utilizing a combination of RAG technique, a fine-tuned specialized model and a general LLM (for which we picked GPT-4o; [Methods](/articles/s41551-025-01463-z#Sec13)).

To enhance the Q&A mode’s capacity to ‘think’ like a scientist for problem solving, we sought to train a specialized language model using real scientific discussions among domain experts. The fine-tuned model is used as one of the multiple sources of knowledge for the Q&A mode (Fig. [4e](/articles/s41551-025-01463-z#Fig4)). To this end, we collected 11 years of open-forum discussions from a public Google Discussion Group on CRISPR gene-editing, starting from 2013 (Supplementary Note [B](/articles/s41551-025-01463-z#MOESM1)). The discussion group involved a diverse cohort of scientists worldwide. This dataset, comprising ~4,000 discussion threads, was curated into an instructional dataset with over 3,000 question-and-answer pairs (Supplementary Note [B](/articles/s41551-025-01463-z#MOESM1)). Using this dataset, we fine tuned an 8-billion-parameter LLM on the basis of the Llama3-instruct model[58](#ref-CR58 "Introducing Meta Llama 3: the most capable openly available LLM to date. Meta Blog
                https://ai.meta.com/blog/meta-llama-3/

               (2024)."),[59](#ref-CR59 "Vaswani, A. et al. Attention is all you need. Adv. Neural Inf. Process. Syst. 30, 5998–6008 (2017)."),[60](#ref-CR60 "Loshchilov, I. et al. Decoupled Weight Decay Regularization (ICLR, 2019)."),[61](#ref-CR61 "Hu, E. et al. LoRA: Low-Rank Adaptation of Large Language Models (ICLR, 2022)."),[62](#ref-CR62 "Dettmers, T., Pagnoni, A., Holtzman, A. & Zettlemoyer, L. QLORA: efficient finetuning of quantized LLMs. In Proc. 37th International Conference on Neural Information Processing Systems (eds Oh, A. et al.) 441 (NeurIPS, 2023)."),[63](#ref-CR63 "Zhou, C. et al. Lima: Less is more for alignment. In Proc. 37th International Conference on Neural Information Processing Systems (eds Oh, A. et al.) 2400 (NeurIPS, 2023)."),[64](#ref-CR64 "Delvin, J., Chang, M.-W., Lee, K. & Toutanova, K. BERT: pre-training of deep bidirectional transformers for language understanding. Proc. Conf. NAACL-HLT 2019, 4171–4186 (2019)."),[65](/articles/s41551-025-01463-z#ref-CR65 "Liu, Z., Lin, W., Shi, Y. & Zhao, J. RoBERTa: A Robustly Optimized BERT Pretraining Approach. In Proc. 20th Chinese National Conference on Computational Linguistics (eds Li, S. et al.) 1218–1227 (Chinese Information Processing Society of China, 2021)."). The fine-tuned model, which we call CRISPR-Llama3, has improved abilities in gene-editing questions, outperforms the baseline model on basic questions by a moderate 8% and on real-world research questions by ~20% (Supplementary Figs. [2](/articles/s41551-025-01463-z#MOESM1) and [3](/articles/s41551-025-01463-z#MOESM1)). We integrate this fine-tuned LLM into the Q&A mode as a ‘brainstorming source’, enabling the agent to generate ideas like a human scientist and provide a second opinion for difficult queries (Fig. [4e](/articles/s41551-025-01463-z#Fig4)).

To assess the performance of the Q&A mode, we used the Gene-editing bench Q&A test set (Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). The test questions encompass basic gene-editing knowledge, experimental troubleshooting, CRISPR application in various biological systems, ethics and safety[66](/articles/s41551-025-01463-z#ref-CR66 "Frequently Asked Questions About CRISPR (Innovative Genomics Institute, 2025)."),[67](/articles/s41551-025-01463-z#ref-CR67 "Beumer, J. et al. CRISPR: questions and answers. STAR Protoc.
                https://star-protocols.cell.com/protocols/2555

               (2023)."). We prompted CRISPR-GPT, GPT-3.5-turbo and GPT-4o to generate responses to test questions. Three human experts scored the answers in a fully blinded setting. The test demonstrated that the Q&A mode outperformed baseline LLMs in accuracy, reasoning and conciseness, with improvement of 12%, 15% and 32%, respectively, versus GPT-4o (Fig. [4f](/articles/s41551-025-01463-z#Fig4)). Human evaluators observed that general-purpose LLMs sometimes make factual errors and tend to provide extensive answers not all of which are relevant to the questions. For example, one question was about solving cell growth issues in an experiment where a scientist performed Cas9 editing followed by single-cell sorting using MCF-7 cells. For this question, the Q&A mode provided a concise, accurate summary of potential reasons and actionable solutions. In contrast, GPT-4o responded with a long list of 9 factors/options, but at least 2 of them were not applicable to MCF-7 cells (Extended Data Fig. [2](/articles/s41551-025-01463-z#Fig8)). This and other examples (Extended Data Figs. [3](/articles/s41551-025-01463-z#Fig9) and [4](/articles/s41551-025-01463-z#Fig10)) showcase the advantage of the CRISPR-GPT Q&A mode. Overall, evaluation results confirmed that the multisource Q&A mode is better at answering advanced research questions about gene editing.

### CRISPR-GPT excels in human–AI collaboration

To further evaluate the human user experience of CRISPR-GPT, we assembled a panel of eight gene-editing experts to assess the agents’ performance for end-to-end experiment covering all 22 individual tasks (Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1) and [3](/articles/s41551-025-01463-z#MOESM1) demos). The experts were asked to rate their experiences in four dimensions: accuracy, reasoning and action, completeness and conciseness (see Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1) for detailed rubrics). CRISPR-GPT demonstrated improved accuracy and strong capabilities in reasoning and action, whereas general LLMs, such as GPT-4o, often included errors and were prone to hallucination (Fig. [5a,b](/articles/s41551-025-01463-z#Fig5)).

**Fig. 5: CRISPR-GPT outperforms general-purpose LLM for gene-editing research in human user experiences.**

[![figure 5](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig5_HTML.png)](/articles/s41551-025-01463-z/figures/5)

**a**, Human user experience: evaluation of CRISPR-GPT for end-to-end gene-editing copiloting. Human experts scored performances from 1 (lowest) to 5 (highest). See detailed procedure and rubrics in Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1) (full chat history and video demo provided in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). **b**, Human user experience: evaluation results breakdown by major gene-editing tasks. Data shown are the mean ± s.d. (*n* = 8 per group). **c**, User observations on the strengths and limitations of CRISPR-GPT compared to baseline LLMs.

[Full size image](/articles/s41551-025-01463-z/figures/5)

Highlighted by human evaluators’ observations (Fig. [5c](/articles/s41551-025-01463-z#Fig5)), the CRISPR-GPT agent provides users with more accurate, concise and well-rounded instructions to execute the planned experiments. The ability of CRISPR-GPT to perform specialty gene-editing tasks, such as exon-selected gRNA design, customized off-target prediction and automated sequencing data analysis, reinforced its advantage versus general-purpose LLMs. This is confirmed by the task-specific evaluation results (Fig. [5b](/articles/s41551-025-01463-z#Fig5)). Despite its strengths, CRISPR-GPT struggled with complex requests and rare biological cases, highlighting areas for improvement (limitations in Supplementary Note [D](/articles/s41551-025-01463-z#MOESM1)).

### Real-world demonstration for fully AI-guided gene editing

To showcase and validate CRISPR-GPT’s ability as an AI co-pilot to general biologists, we enlisted two junior researchers unfamiliar with gene editing. They used CRISPR-GPT in two real-world experiments: (1) designing and conducting a multigene knockout and (2) epigenetic editing, from scratch.

In the first experiment, the junior researcher conducted gene knockouts in the human A549 lung adenocarcinoma cell line, targeting four genes involved in tumour survival and metastasis: *TGFβR1*, *SNAI1*, *BAX* and *BCL2L1* (Fig. [6](/articles/s41551-025-01463-z#Fig6)). The experiment was designed from scratch with CRISPR-GPT (Fig. [6a](/articles/s41551-025-01463-z#Fig6)). On the basis of the user–AI interaction, enAsCas12a was selected for its multitarget editing capability and low off-target effects. For delivery, CRISPR-GPT recommended lentiviral transduction for stable Cas and gRNA expression. The gRNAs for the four target genes were designed through CRISPR-GPT. Furthermore, CRISPR-GPT provided step-by-step protocols for gRNA cloning, lentivirus production and viral delivery into A549 cells. To validate the editing, the researcher followed CRISPR-GPT’s next-generation sequencing (NGS) protocol, using assay primers designed via the integrated Primer3 tool. After generating the NGS data, the raw sequencing files were uploaded into CRISPR-GPT for automated analysis through the CRISPResso2 pipeline. The analysis reports, sent directly via email, summarized the editing outcomes and showed consistently ~80% high efficiency across all target genes (Fig. [6b](/articles/s41551-025-01463-z#Fig6), Supplementary Demo Video [3](/articles/s41551-025-01463-z#MOESM7), user interactions summarized in Supplementary Table [4](/articles/s41551-025-01463-z#MOESM1), full chat history listed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). To further assess the biological phenotypes of *TGFβR1* and *SNAI1* knockout in A549 cells, the researcher conducted an epithelial–mesenchymal transition (EMT) induction experiment by treating A549 cells with TGFβ (Fig. [6c](/articles/s41551-025-01463-z#Fig6) and [Methods](/articles/s41551-025-01463-z#Sec13)). The qPCR results revealed that the knockout A549 cell lines (A549 TGFβR1 KO and A549 SNAI1 KO) showed up to 9-fold reduction in CDH1 expression change and up to 34-fold reduction in VIM expression change, which are both key marker genes in the EMT process. This confirms the biological role of TGFβR1 and SNAI1 signalling in driving EMT progression (a crucial driver of metastasis) in lung cancer cells (Fig. [6d](/articles/s41551-025-01463-z#Fig6)).

**Fig. 6: Wet-lab demonstrations of CRISPR-GPT in knockout and activation experiments.**

[![figure 6](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41551-025-01463-z/MediaObjects/41551_2025_1463_Fig6_HTML.png)](/articles/s41551-025-01463-z/figures/6)

**a**, The full workflow of CRISPR-GPT-guided knockout experiment of *TGFβR1*, *SNAI1*, *BAX1* and *BCL2L1* through multiple rounds of human–AI interaction (*TGFβR1* knockout is shown as an example, see Supplementary Demo Video [3](/articles/s41551-025-01463-z#MOESM7) and full chat history listed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). Panel was originally created with [BioRender.com/iim1m2t](https://BioRender.com/iim1m2t) and processed. **b**, Editing efficiencies for *TGFβR1*, *SNAI1*, *BAX1* and *BCL2L1* measured via NGS, analysed using CRISPResso2 and CRISPR-GPT. **c**, Schematic of the EMT induction process via TGFβ treatment ([Methods](/articles/s41551-025-01463-z#Sec13)). Panel was originally created with [BioRender.com/iim1m2t](https://BioRender.com/iim1m2t) and processed. **d**, Functional outcomes of *TGFβR1* and *SNAI1* knockout in A549 cells after EMT induction by TGFβ. qPCR analysis shows reduced expression changes in EMT marker genes (*CDH1*, *VIM*) in A549 *TGFβR1* and *SNAI1* knockout cells compared with A549 wildtype (WT) cells, confirming successful knockouts of *TGFβR1*/*SNAI1*. Data shown are the mean ± s.d. (*n* = 3 per group). One-way analysis of variance (ANOVA) was used to calculate statistics. \*\*\*\**p* < 0.0001. **e**, Simplified workflow of a beginner researcher activating *NCR3LG1* and *CEACAM1* expression through multiround interactions with CRISPR-GPT (full chat history listed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). **f**, Editing outcomes of *NCR3LG1* and *CEACAM1* activation using CRISPR-GPT-designed sgRNAs, measured via flow cytometry ([Methods](/articles/s41551-025-01463-z#Sec13)).

[Full size image](/articles/s41551-025-01463-z/figures/6)

In the second experiment, the junior researcher performed epigenetic editing to activate two genes involved in cancer immunotherapy resistance in a human melanoma model cell line (Fig. [6e](/articles/s41551-025-01463-z#Fig6), user interactions summarized in Supplementary Table [4](/articles/s41551-025-01463-z#MOESM1), full chat history listed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). CRISPR-GPT guides the researcher through the full workflow: identifying the most suitable CRISPR activation system, selecting an appropriate delivery method for A375 cells, designing dCas9 gRNAs (three gRNAs per gene) and generating protocols for validating editing outcomes. After editing was completed, measurements of target protein expression level confirmed successful activation of both genes, with up to 56.5% efficiency for *NCR3LG1*, and 90.2% efficiency for *CEACAM1*, when comparing gRNA-edited groups vs negative control gRNAs (Fig. [6f](/articles/s41551-025-01463-z#Fig6)).

Overall, CRISPR-GPT enabled successful completion of the first set of AI-guided gene-editing experiments. Interactions between the researchers and LLM-powered agents led to efficient, accurate and ethically mindful gene-editing on the first attempt, even by users new to the technique.

## Discussion

CRISPR-GPT demonstrates the potential of LLMs to automate and enhance biological research and experiment design and analysis. This AI-guided workflow leverages LLM for reasoning, multi-agent collaboration and scientific discussions for brainstorming, reduces errors, and improves research quality and reproducibility. Despite its current capabilities, CRISPR-GPT has limitations. For example, the agent system relies on high-quality instructions and discussion data from human scientists who have deep knowledge about the biology domain. Such data are hard to collect, creating challenges for further improvements and scaling up. Further, evaluation of such AI tools is generally challenging due to the need to collect substantial feedback from human biologists. For another example, the current gRNA design step mainly supporting human and mouse targets could be further expanded.

Technologies such as CRISPR-Cas9 pose potential ethical and safety risks, including potential misuse for dual purposes, which can be exemplified with AI tools[68](/articles/s41551-025-01463-z#ref-CR68 "Doudna, J. A. The promise and challenge of therapeutic genome editing. Nature 578, 229–236 (2020)."). Altering human genomes raises substantial ethical concerns, particularly with germline cell and embryo editing. Due to these concerns, such editing is illegal in the United States and many other countries. In addition, gene-editing technology could be abused to create bioweapons, such as genetically engineered viruses[69](/articles/s41551-025-01463-z#ref-CR69 "Hoffmann, S. A. et al. Safety by design: biosafety and biosecurity in the age of synthetic genomics. iScience 26, 106165 (2023).").

To mitigate these risks, we augment CRISPR-GPT with an additional layer of safety mechanism to defend against malicious dual uses. Following the guidelines given in a moratorium[70](/articles/s41551-025-01463-z#ref-CR70 "Lander, E. S. et al. Adopt a moratorium on heritable genome editing. Nature 567, 165–168 (2019).") on heritable genome editing, CRISPR-GPT ensures that users cannot bypass the step of specifying the organism they are editing. If the target is human tissue or organs, the system triggers the following steps: (1) displays a warning note when proceeding with human gene-editing experiments; (2) provides a link to the international moratorium with an explanatory note; and (3) asks users to confirm that they understand the risks and have read the international guidelines before proceeding. The agent also checks if the user request involves editing of human germline cells or dangerous, pathogenic viruses (Supplementary Note [D](/articles/s41551-025-01463-z#MOESM1) and Data [2](/articles/s41551-025-01463-z#MOESM4)). If such a risk is identified, it will trigger an error message and stop proceeding (Extended Data Fig. [5](/articles/s41551-025-01463-z#Fig11) for examples of the risk mitigation tests).

Other concerns are related to user data privacy issues, especially when human genome sequence information might be exchanged by using AI tools. We follow the data privacy and HIPAA privacy rule in healthcare[71](/articles/s41551-025-01463-z#ref-CR71 "Summary of the HIPAA Privacy Rule
                https://www.hhs.gov/hipaa/for-professionals/privacy/laws-regulations/index.html

               (US Department of Health and Human Services, 2008)."). Although genome-scale sequences are fundamentally linked to identities, DNA segments of up to 20-bp length are considered safe and not able to identify human identity[72](/articles/s41551-025-01463-z#ref-CR72 "Naveed, M. et al. Privacy in the genomic era. ACM Comput. Surv. 48, 6 (2015).").

CRISPR-GPT includes functionalities to prevent sharing of identifiable private human/patient sequences with public LLM models. Our solution involves two key measures. First, the system would never store any identifiable long genome sequence in the server that would potentially reveal patient private information. Second, a filter is implemented to detect any sequence of 20 or more A/T/G/C/U bases in prompts before sending them to external LLMs. If detected, the agent raises an error with a warning note, asking the user to manually remove the sequence from the input. This prevents the leakage of sensitive information to external models and tools (Extended Data Fig. [5](/articles/s41551-025-01463-z#Fig11)).

Looking ahead, the uses of CRISPR-GPT could be further expanded by connecting to latest advances in genome/protein foundation models, plasmid design tools and other machine learning models, to enable experiment design and analysis tasks beyond gene editing. In addition, the integration of CRISPR-GPT with automated laboratory platforms and robotics holds immense promise. By bridging computational design, analysis and physical execution, researchers could leverage the agent’s expertise to orchestrate end-to-end automated experiments, minimizing manual intervention and accelerating the pace of discovery.

## Methods

### Large language model-powered autonomous agent

CRISPR-GPT consists of the following 4 core components (Fig. [1](/articles/s41551-025-01463-z#Fig1)): LLM Planner, Tool providers, Task executors, and the LLM User-proxy agent that serve as the interface with users for taking inputs and communicating outputs. Each component can be viewed as an LLM-powered single agent with relatively simple functionality, and the overall system functions via multi-agent interaction. These single agents leverage general-purpose LLMs, such as GPT-4o (used in all four core agents unless otherwise specified, such as when comparing GPT-4 vs GPT-3.5, where we use GPT-4o and GPT-3.5), as their base model to handle a wide range of tasks. The LLMs rely on carefully designed prompts to guide their behaviour and interactions, with example prompts provided in Supplementary Note [E](/articles/s41551-025-01463-z#MOESM1).

### The Task executor operates as state machines, providing robust decomposition and progress contro

A total of 22 tasks (summarized in Supplementary Table [1](/articles/s41551-025-01463-z#MOESM1)) have been implemented, each decomposed into subgoals, with states providing instructions and guiding users through decision-making via multiple rounds of textual interaction. A central management class tracks the current state, task queue, and memory for state outputs and execution history. State transitions occur sequentially, or based on conditional logic from execution results as needed. States process user input and generate structured outputs containing the status, reasoning and response, which are stored continuously to ensure continuity across tasks. This framework supports both predefined workflows (Meta mode) and dynamically generated task sequences (Auto mode), offering flexibility, reliability and robust error handling for executing CRISPR-GPT tasks.

In Meta mode, the Task executor follows predefined workflows that cover complete pipelines for four meta-tasks, each corresponding to a major type of gene-editing experiment. In Auto mode, the LLM Planner dynamically generates a customized sequence of tasks on the basis of the user’s meta-request. The Task executor then constructs and executes the workflow by chaining the state machines of the corresponding tasks into a larger state machine, enabling seamless and automated execution of complex gene-editing pipelines.

### The Tool provider connects the Task executor with external application programming interfaces (APIs)

To connect language models with external functionalities[73](#ref-CR73 "OpenAI. ChatGPT plugins.
                https://openai.com/blog/chatgpt-plugins

               (2023)."),[74](#ref-CR74 "Shen, Y. et al. HuggingGPT: solving AI tasks with ChatGPT and its friends in hugging face. In Advances in Neural Information Processing Systems (eds Oh, A. et al.) Vol. 36, 38154–38180 (Curran Associates, Inc., 2023)."),[75](#ref-CR75 "Wang, G. et al. Voyager: an open-ended embodied agent with large language models. Preprint at
                https://doi.org/10.48550/arXiv.2305.16291

               (2023)."),[76](#ref-CR76 "Hao, S., Liu, T., Wang, Z. & Hu, Z. ToolkenGPT: augmenting frozen language models with massive tools via tool embeddings. In Advances in Neural Information Processing Systems (eds Oh, A. et al.) Vol. 36, 45870–45894 (Curran Associates, Inc., 2023)."),[77](/articles/s41551-025-01463-z#ref-CR77 "Patil, S. G., Zhang, T., Wang, X. & Gonzalez, J. E. Gorilla: large language model connected with massive APIs. In Advances in Neural Information Processing Systems (eds Globerson, A. et al.) Vol. 37, 126544–126565 (Curran Associates, Inc., 2024)."), the system needs to (1) analyse the current situation and judge whether it is suitable to call an external tool; and (2) know what kinds of tools are available and choose the best from them. Instead of directly exposing the interfaces of the APIs to LLMs, in CRISPR-GPT, we wrap the usage of APIs inside the states and expose more user-friendly and LLM-friendly textual interfaces through hand-written instructions and responses. In plain words, we are teaching users (human agents and LLM User-proxy agents) to use the tools. The tools include Google web search, Google Scholar search, literature retrieval and bioinformatic tools such as Primer3 (ref. [52](/articles/s41551-025-01463-z#ref-CR52 "Untergasser, A. et al. Primer3—new capabilities and interfaces. Nucleic Acids Res. 40, e115 (2012).")), CRISPRitz[53](/articles/s41551-025-01463-z#ref-CR53 "Cancellieri, S., Canver, M. C., Bombieri, N., Giugno, R. & Pinello, L. CRISPRitz: rapid, high-throughput and variant-aware in silico off-target site identification for CRISPR genome editing. Bioinformatics 36, 2001–2008 (2020).") and CRISPResso2 (ref. [57](/articles/s41551-025-01463-z#ref-CR57 "Clement, K. et al. CRISPResso2 provides accurate and rapid genome editing sequence analysis. Nat. Biotechnol. 37, 224–226 (2019).")).

### The LLM Planner automatically plans gene-editing experiments on the basis of the user’s request

LLMs such as GPT-4, Gemini and Claude serve as the reasoning core of the LLM-powered agent to solve real-world decision-making problems. Our LLM Planner operates on the basis of two key components: (1) the user query and (2) a predefined table containing comprehensive descriptions and interdependency information for all available tasks (example in Supplementary Note [E](/articles/s41551-025-01463-z#MOESM1)). Using the ReAct prompting technique, the LLM is prompted to output a chain-of-thought reasoning path along with the final action from the plausible action set (Fig. [2a](/articles/s41551-025-01463-z#Fig2)). On the basis of LLM’s internal knowledge, combined with our manually written task descriptions and decomposition instructions, the Planner analyses the user’s request, intelligently decomposes it into an ordered list of tasks, and ensures that the dependencies between tasks are respected (detailed prompt format provided in Supplementary Note [E](/articles/s41551-025-01463-z#MOESM1)). Once the decomposition is complete, the corresponding state machines are automatically chained together to execute all tasks in the appropriate sequence. For robustness, we prevent the LLM from dynamically adding or deleting tasks (state machines) during execution. However, we acknowledge that enabling dynamic task management is an important step towards developing a more intelligent science AI agent and leave this for future work.

### LLM User-proxy agent automatically interacts with the Task executor on the basis of the meta-request

The LLM User-proxy agent automatically interacts with the Task executor on the basis of the meta-request. Central to our system, this agent serves as an intermediary between the user and a state machine derived from an initial task decomposition step—breaking down the gene-editing process into a structured sequence of actions and decisions. At each step, the state machine presents a current state to the LLM agent, which includes a task description and specifies any required input. This input varies by task type and may include general experimental context (for example, “I hope to design 4 sgRNAs targeting human TGFβR1”) or a specific Cas system (for example, enCas12a), as shown in Supplementary Fig. [1b](/articles/s41551-025-01463-z#MOESM1).

The LLM User-proxy agent interprets the current state and makes informed decisions on the user’s behalf. It integrates multiple sources of information, including:

1. 1.

   Instructions from the state machine,
2. 2.

   User requests,
3. 3.

   Session interaction history, and
4. 4.

   Results from external computational tools.

This synthesized information is formatted into a structured prompt for the agent, which determines the most appropriate next action. For instance, when designing a CRISPR experiment, the agent might combine user input about a target gene with computational results to suggest sgRNA candidates.

While the User-proxy agent operates autonomously, user oversight remains essential. Users are encouraged to monitor task progression and intervene as needed to correct errors or misinterpretations, preserving the integrity of the gene-editing design (Supplementary Fig. [1a](/articles/s41551-025-01463-z#MOESM1)).

This approach fosters a collaborative synergy between human expertise and AI. By leveraging the LLM agent’s reasoning ability, we enable a more efficient, accurate and user-friendly design experience. The sequential decision-making framework streamlines execution while ensuring that user input remains central to experiment planning.

### Delivery method selection agent

Our approach mirrors the thought process of human gene-editing experts to identify the most appropriate delivery method on the basis of the user’s specific biological system. The workflow is illustrated in Fig. [4a](/articles/s41551-025-01463-z#Fig4). It begins by instructing the LLM to extract key biological terms from the user’s natural language request. These terms provide insight into the biological context of the experiment. The LLM is then tasked with accessing up-to-date information using a Google Search API to gather additional context about the biological system in the user request.

On the basis of the combined information from the user’s request and external data, the LLM categorizes the system into one of 7 major biological categories:

1. 1.

   Mammalian in vivo
2. 2.

   Mammalian embryos
3. 3.

   Mammalian primary cells or stem cells ex vivo
4. 4.

   Mammalian cell lines with strong evidence of high-efficiency transfection
5. 5.

   Mammalian cell lines or organoids without strong evidence of high-efficiency transfection
6. 6.

   Human in vivo or human embryos
7. 7.

   Bacteria, viruses and other organisms

These categories encompass the majority of biological systems relevant to CRISPR delivery. For each category, we curated 1–3 delivery methods on the basis of human experts’ knowledge, which represent the most commonly used CRISPR delivery strategies.

To further tailor the recommendations to the user’s specific scenario, the agent system conducts a Google Scholar search to identify relevant peer-reviewed literature. The search is guided by the key terms extracted from the user’s request. From the search results, the top 10 relevant papers are ranked by citation count, providing a quantitative measure for prioritizing the potential delivery options within each biological category.

While citation numbers are not a definitive metric for determining the most appropriate delivery method, they offer a useful reference point. This approach helps to present well-informed recommendations along with relevant literature to the user.

### Guide RNA design agent

Designing sgRNAs is a critical challenge in CRISPR editing, as it directly impacts editing efficiency. While many sgRNA design tools (web-based and software packages) exist, they typically follow shared design principles and use metrics, such as on-target/off-target scores, exon number and cut position, to rank candidates. We identified two main user challenges: (1) finding a trustworthy sgRNA design source and (2) efficiently selecting sgRNAs that meet specific criteria without manually evaluating every metric.

To address these issues, we leveraged pre-designed sgRNA tables from CRISPick, a widely used and validated library from the Broad Institute. We combined this with the reasoning and action (ReAct) capabilities of LLMs to process user-driven table queries. Our agent executes a series of actions step by step to generate outputs, akin to the recently described ‘chain-of-table’ methodology[15](/articles/s41551-025-01463-z#ref-CR15 "O’Donoghue, O. et al. BioPlanner: automatic evaluation of LLMs on protocol planning in biology. In Proceedings of the 2023 Conference on Empirical Methods in Natural Language Processing (eds Bouamor, H., Pino, J. & Bali, K.) 2676–2694 (Association for Computational Linguistics, 2023).").

The agent system can choose from four key functions:

* SELECT: Retrieves rows where the specified column matches the given value.
* BETWEEN: Selects rows where the specified column’s values fall within a specified range (inclusive).
* ORDERBY: Orders the table on the basis of values in a specified column.
* TOP: Returns the top *N* rows of the table.

These can be expanded in the future via human or LLM-driven suggestions. The agent extracts relevant parameters from both the user request and table, applies the functions, and returns pre-designed sgRNAs with associated metadata through a visual table and download link.

We also developed an optional exon suggestion module for CRISPR knockout design. Since sgRNAs targeting non-essential regions can be less effective, we hypothesized that LLMs could use their broad knowledge base to suggest functionally important exons. For example, targeting the BD1/BD2 domains has been shown to effectively disrupt BRD4 function[44](/articles/s41551-025-01463-z#ref-CR44 "Wang, Z. et al. Chain-of-table: evolving tables in the reasoning chain for table understanding. In Proc. 12th International Conference on Learning Representations (OpenReview, 2024)."). We prompt the LLM to reason through gene function and recommend candidate exons (Extended Data Fig. [1](/articles/s41551-025-01463-z#Fig7)), which are then incorporated into table queries.

To our knowledge, no existing sgRNA design tools integrate gene functional domains, making this exon suggestion feature a valuable reference. However, we note that its performance may be limited for genes with sparse literature or minimal online information.

### Q&A mode

General-purpose LLMs often lack sufficient understanding of advanced biology. As outlined in Supplementary Note [A](/articles/s41551-025-01463-z#MOESM1), we identified key failure modes: (1) information hallucination, (2) outdated CRISPR knowledge, (3) absence of peer-reviewed sources and (4) poor alignment with user-specific problem-solving needs. To overcome these challenges, CRISPR-GPT’s Q&A mode employs a multisource system for answering advanced biology questions (Fig. [4e](/articles/s41551-025-01463-z#Fig4)). Upon receiving a query, it synthesizes information from three sources:

1. 1.

   Fine-tuned CRISPR-LLama: Trained on human discussion threads from a CRISPR-focused Google Group, this model improves problem-solving and troubleshooting beyond the baseline (Supplementary Note [B](/articles/s41551-025-01463-z#MOESM1)).
2. 2.

   RAG-based literature retrieval (Tool provider agent): This accesses a curated, expert-selected CRISPR literature database (~50 key papers chosen for impact and recency; Supplementary Fig. [4](/articles/s41551-025-01463-z#MOESM1) and Note [F](/articles/s41551-025-01463-z#MOESM1)). Using OpenAI Embeddings and FAISS, both database entries and user queries are embedded into semantic vectors. The top *k* (*k* = 4) passages are retrieved by cosine similarity, ranked and summarized to guide the model’s response.
3. 3.

   General-purpose LLM: For example, ChatGPT or LLama, used as an additional source.

### Extendibility of CRISPR-GPT

Given that CRISPR-GPT has a modular multi-agent architecture, integrating new tools and functions into the existing system is easy and training free. To add a new tool/function, the procedure is as follows:

1. (1)

   Tool wrapping: Develop specific code to encapsulate the tool’s functionality within a state machine, which we call a Tool provider agent. This wrapper presents user-friendly and LLM-friendly textual interfaces through carefully crafted instructions and responses.
2. (2)

   Meta mode integration: If we want to add the tool to be used in the Meta mode, we add the entry state of the new state machine to appropriate positions within the relevant predefined workflow.
3. (3)

   Auto mode integration: Register the entry state of the new tool’s state machine in the task decomposition table. This ensures that during task decomposition, the Planner agent becomes aware of the new tool and can incorporate it into its decision-making process.

### Performance assessment of CRISPR-GPT

#### Benchmark dataset

We compiled Gene-editing bench, a collection of test questions and answers for evaluating AI tools’ capabilities for CRISPR experimental design, with a total of 288 unique entries covering four topics:

1. 1.

   Gene-editing planning: we compiled a total of 50 test cases and answers curated by consensus of human gene-editing experts.
2. 2.

   CRISPR gRNA design: 50 test cases with pre-compiled answers by human experts.
3. 3.

   Gene-editing delivery method selection: 50 test cases covering a range of biological systems and major experiment types. For each test case, we asked human experts to rank the available delivery method and report the consensus ranking as answer.
4. 4.

   Gene-editing Q&A: 138 questions and answers, filtered for errors or issues, compiled from both public sources and human experts.

### Validation of individual gene-editing agents

Using this benchmark dataset, we evaluated key functions of the CRISPR-GPT agent system:

1. 1.

   Planning evaluation: For each query, we generated three batches of subtask lists using CRISPR-GPT and compared them to ground truth using accuracy, precision, recall and F1 scores. Task ordering was assessed via normalized Levenshtein distance. We also tested GPT-4o and GPT-3.5-turbo for comparison, evaluating the models’ ability to plan and sequence gene-editing tasks.
2. 2.

   Delivery method selection: For each test case, CRISPR-GPT (with and without literature search), GPT-3.5-turbo and GPT-4-turbo proposed primary and secondary delivery methods. Responses were scored against ground truth (primary: weight 2; secondary: weight 1), summed across categories to assess each model’s ability to suggest delivery methods across biological systems.
3. 3.

   Guide RNA design evaluation: CRISPR-GPT generated gRNA design function lists and parameters, which were compared to ground truth to evaluate function selection, order and parameter accuracy. We also tested GPT-4 and GPT-3.5-turbo, assessing their ability to interpret user intent and produce effective design strategies.
4. 4.

   Q&A mode evaluation: We selected 40 questions and prompted CRISPR-GPT, GPT-3.5-turbo and GPT-4 to generate answers. Three human experts scored responses across four aspects in a blinded setup, and average scores were used to determine final performance, evaluating the models’ ability to handle diverse gene-editing questions.

Detailed evaluation procedures for all the above are provided in Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1).

### Human experience evaluation

To holistically evaluate user experiences of the CRISPR-GPT, we invited 8 independent CRISPR human experts to test the agent system via its web surface. Each expert was asked to make one gene-editing request under the Meta mode and two gene-editing requests under the Auto mode. More details on the evaluation procedures are given in Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1). In addition, we also provide a total of 20 full chat history demos from these tests in Supplementary Data [1](/articles/s41551-025-01463-z#MOESM3) (details listed in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)).

### Real-world applicability of CRISPR-GPT via wet lab demonstrations

To evaluate the real-world applicability of CRISPR-GPT, we conducted two independent wet lab demonstrations:

1. 1.

   Beginner Researcher 1: We invited an independent junior PhD scientist unfamiliar with the CRISPR field to perform CRISPR gene-editing experiments using CRISPR-GPT via human–agent collaboration. The researcher applied CRISPR-GPT to execute a gene knockout experiment as part of a cancer research project. The agent provided step-by-step guidance throughout the process (video demonstrations are available in Supplementary Demo Videos [1](/articles/s41551-025-01463-z#MOESM5)–[3](/articles/s41551-025-01463-z#MOESM7), full chat history in Supplementary Data [1](/articles/s41551-025-01463-z#MOESM3), and details in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). The results were validated through next-generation sequencing and functional assays.
2. 2.

   Beginner Researcher 2: An undergraduate student, also unfamiliar with the CRISPR field, was invited to perform gene-editing experiments through collaboration with CRISPR-GPT. The student implemented CRISPR activation in a cancer immunology research project, with stepwise guidance provided by the agent (full chat history provided in Supplementary Data [1](/articles/s41551-025-01463-z#MOESM3), details in Supplementary Table [2](/articles/s41551-025-01463-z#MOESM1)). The results were validated through antibody staining and FACS sorting.

### Cell line and cell culture

A375 and A549 cells were cultured in DMEM medium (high glucose, GlutaMAX; 10-569-044 Gibco) with 10% fetal bovine serum (FBS; 100-106, Gemini Bio), 100 U ml−1 penicillin and 100 μg ml−1 streptomycin (15140-122 Gibco). Cells were maintained at 37 °C in a humidified atmosphere with 5% CO2.

### CRISPR RNA cloning

Cloning of sgRNAs was carried out using BbsI or Esp3I (R3539, R0734 NEB) through a Golden Gate assembly into a lentiviral backbone. The constructs were sequence-verified via Sanger sequencing using a U6 sequencing primer (5’-GACTATCATATGCTTACCGT-3’).

### Lentivirus packaging and transduction

Lentivirus production was performed by co-transfecting the assembled lentiviral vector with VSV-G envelope and Delta-Vpr packaging plasmids into HEK-293T cells using PEI transfection reagent (765090, Sigma-Aldrich). Supernatants were collected at 48 h post transfection. A375 and A549 cells were transduced at low multiplicity of infection (MOI) with 8 μg ml−1 polybrene using a spin infection method at 1,000 × *g* for 45 min. After 24 h, cells were selected with 1 μg ml−1 puromycin to establish stable cell lines.

### Genomic DNA extraction, PCR and sequencing

Genomic DNA was extracted from selected cells at 7 days post transfection using QuickExtract DNA Extraction Solution (LGCQE09050, Lucigen) following manufacturer instructions. Targeted loci were amplified via PCR using Phusion Flash High-Fidelity PCR Master Mix (F548L, ThermoFisher) with primers containing Illumina sequencing adapters. Paired-end reads (150 bp) were generated using the Illumina MiSeq platform.

PCR primers:

* TGFβR1: Forward: AGATAGAGGGTACTACGTTGAAAGACT; reverse: AAAAAAGTCTTTCAACGTAGTACCCTCT
* SNAI1: Forward: AGATCAGTTGAAGGCCTTTCGAGCCTG; reverse: AAAACAGGCTCGAAAGGCCTTCAACTG
* BAX: Forward: AGATATCCAGGATCGAGCAGGGCGAAT; reverse: AAAAAATTCGCCCTGCTCGATCCTGGAT
* BCL2L1: Forward: AGATACGCACAGTGCCCCGCCGAAGGA; reverse: AAAATCCTTCGGCGGGGCACTGTGCGT

### TGFβ treatment

For optimal EMT induction, cells were seeded at a density of 750,000 cells per 100 mm tissue culture plate and incubated for 24 h. The medium was then replaced with 2% FBS DMEM for an additional 24 h. Cells were subsequently treated with 5 ng ml−1 TGFβ (240-B/CF, R&D) in 2% FBS DMEM for 7 days. To ensure consistent cell density during the treatment, cells were reseeded at the same density every 2 days.

### Quantitative PCR

Total RNA was extracted using the Direct-zol RNA Purification kit ([R2051](https://www.zymoresearch.com/collections/direct-zol-rna-kits/products/direct-zol-rna-miniprep-kits), Zymo Research) according to manufacturer instructions. Complementary DNA synthesis and qPCR were performed using the Power SYBR Green RNA-to-CT 1-Step kit (4389986, ThermoFisher) on a BioRad CFX384 system (BioRad). Gene expression was quantified using specific primers for *CDH1* (forward: CTG AGG ATG GTG TAA GCG ATG; reverse: GTC TGT CAT GGA AGG TGC TC) and *VIM* (forward: GTG AAT CCA GAT TAG TTT CCC TCA; reverse: CAA GAC CTG CTC AAT GTT AAG ATG). Expression levels were normalized to appropriate housekeeping genes.

### Flow cytometry (FACS) analysis

To assess the expression of NCR3LG1 and CEACAM1, cells were stained with B7-H6 monoclonal antibody (JAM1EW), PE (eBioscience) (1:100 dilution) for NCR3LG1 and anti-CD66a/c/e mouse monoclonal antibody (PE [Phycoerythrin], clone ASL-32, 1:100 dilution) for CEACAM1. Staining was performed following manufacturer guidelines, and data were acquired using a CytoFLEX analyser. Flow cytometry data were analysed using standard software. Detailed gating strategies are shown in Supplementary Fig. [5](/articles/s41551-025-01463-z#MOESM1).

### Reporting summary

Further information on research design is available in the [Nature Portfolio Reporting Summary](/articles/s41551-025-01463-z#MOESM2) linked to this article.

## Data availability

The main data supporting the results in this study are available within the paper and its [Supplementary Information](/articles/s41551-025-01463-z#MOESM1). Additional data are deposited in GitHub at <https://github.com/cong-lab/crispr-gpt-pub> (ref. [78](/articles/s41551-025-01463-z#ref-CR78 "Qu, Y. et al. CRISPR-GPT: an LLM agent for automated design of gene-editing experiments. GitHub
                https://github.com/cong-lab/crispr-gpt-pub

               (2024).")).

## Code availability

Because of safety concerns, data, code and prompts will not be fully released to the public until the development of US regulations in the field of artificial intelligence and its scientific applications. While the full code is not freely available, it has been peer reviewed. We will also release a light version in GitHub at <https://github.com/cong-lab/crispr-gpt-pub> (ref. [78](/articles/s41551-025-01463-z#ref-CR78 "Qu, Y. et al. CRISPR-GPT: an LLM agent for automated design of gene-editing experiments. GitHub
                https://github.com/cong-lab/crispr-gpt-pub

               (2024).")).

## References

1. Chowdhery, A. et al. PaLM: scaling language modeling with pathways. *J. Mach. Learn. Res.* **24**, 240 (2023).

   [Google Scholar](http://scholar.google.com/scholar_lookup?&title=PaLM%3A%20scaling%20language%20modeling%20with%20pathways&journal=J.%20Mach.%20Learn.%20Res.&volume=24&publication_year=2023&author=Chowdhery%2CA)
2. Hoffmann, J. et al. An empirical analysis of compute-optimal large language model training. In *Advances in Neural Information Processing Systems* (eds Koyejo, S. et al.) Vol. 35, 30016–30030 (Curran Associates, Inc., 2022).
3. OpenAI et al. GPT-4 technical report. Preprint at <https://doi.org/10.48550/arXiv.2303.08774> (2024).
4. *The Claude 3 Model Family: Opus, Sonnet, Haiku* (Anthropic, 2024).
5. Anil, R. et al. PaLM 2 technical report. Preprint at <https://doi.org/10.48550/arXiv.2305.10403> (2023).
6. Yao, S. et al. ReAct: synergizing reasoning and acting in language models. In *Proc. 11th International Conference on Learning Representations* (OpenReview, 2023).
7. Liu, R. et al. Mind’s eye: grounded language model reasoning through simulation. In *Proc. 11th International Conference on Learning Representations* (OpenReview, 2023).
8. Schick, T. et al. Toolformer: language models can teach themselves to use tools. In *Advances in Neural Information Processing Systems* (eds Oh, A. et al.) Vol. 36, 68539–68551 (Curran Associates, Inc., 2023).
9. Cai, T., Wang, X., Ma, T., Chen, X. & Zhou, D. Large language models as tool makers. In *Proc. 12th International Conference on Learning Representations* (OpenReview, 2024).
10. Yang, C. et al. Large language models as optimizers. In *Proc. 12th International Conference on Learning Representations* (OpenReview, 2024).
11. Li, T. et al. CancerGPT for few shot drug pair synergy prediction using large pretrained language models. *npj Digit. Med.* **7**, 40 (2024).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=38374445)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC10876664)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CancerGPT%20for%20few%20shot%20drug%20pair%20synergy%20prediction%20using%20large%20pretrained%20language%20models&journal=npj%20Digit.%20Med.&volume=7&publication_year=2024&author=Li%2CT)
12. Wu, S. et al. BloombergGPT: a large language model for finance. Preprint at <https://doi.org/10.48550/arXiv.2303.17564> (2023).
13. M. Bran, A. et al. Augmenting large language models with chemistry tools. *Nat. Mach. Intell.* 6, 525–535 (2024).
14. Boiko, D. A., MacKnight, R., Kline, B. & Gomes, G. Autonomous chemical research with large language models. *Nature* **624**, 570–578 (2023).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3sXis1KmsLfI)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=38123806)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC10733136)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Autonomous%20chemical%20research%20with%20large%20language%20models&journal=Nature&volume=624&pages=570-578&publication_year=2023&author=Boiko%2CDA&author=MacKnight%2CR&author=Kline%2CB&author=Gomes%2CG)
15. O’Donoghue, O. et al. BioPlanner: automatic evaluation of LLMs on protocol planning in biology. In *Proceedings of the 2023 Conference on Empirical Methods in Natural Language Processing* (eds Bouamor, H., Pino, J. & Bali, K.) 2676–2694 (Association for Computational Linguistics, 2023).
16. Jinek, M. et al. A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity. *Science* **337**, 816–821 (2012).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC38XhtFOqsb3L)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=22745249)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=A%20programmable%20dual-RNA-guided%20DNA%20endonuclease%20in%20adaptive%20bacterial%20immunity&journal=Science&volume=337&pages=816-821&publication_year=2012&author=Jinek%2CM)
17. Cong, L. et al. Multiplex genome engineering using CRISPR/Cas systems. *Science* **339**, 819–823 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXit1ygtb8%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23287718)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795411)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Multiplex%20genome%20engineering%20using%20CRISPR%2FCas%20systems&journal=Science&volume=339&pages=819-823&publication_year=2013&author=Cong%2CL)
18. Mali, P. et al. RNA-guided human genome engineering via Cas9. *Science* **339**, 823–826 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXit1ygtb0%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23287722)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=RNA-guided%20human%20genome%20engineering%20via%20Cas9&journal=Science&volume=339&pages=823-826&publication_year=2013&author=Mali%2CP)
19. Jinek, M. et al. RNA-programmed genome editing in human cells. *eLife* **2**, e00471 (2013).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23386978)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3557905)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=RNA-programmed%20genome%20editing%20in%20human%20cells&journal=eLife&volume=2&publication_year=2013&author=Jinek%2CM)
20. Doudna, J. A. & Charpentier, E. The new frontier of genome engineering with CRISPR-Cas9. *Science* **346**, 1258096 (2014).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=25430774)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=The%20new%20frontier%20of%20genome%20engineering%20with%20CRISPR-Cas9&journal=Science&volume=346&publication_year=2014&author=Doudna%2CJA&author=Charpentier%2CE)
21. Mali, P., Esvelt, K. M. & Church, G. M. Cas9 as a versatile tool for engineering biology. *Nat. Methods* **10**, 957–963 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXhsFaks7rO)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=24076990)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4051438)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Cas9%20as%20a%20versatile%20tool%20for%20engineering%20biology&journal=Nat.%20Methods&volume=10&pages=957-963&publication_year=2013&author=Mali%2CP&author=Esvelt%2CKM&author=Church%2CGM)
22. Hsu, P. D., Lander, E. S. & Zhang, F. Development and applications of CRISPR-Cas9 for genome engineering. *Cell* **157**, 1262–1278 (2014).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC2cXpslagt7s%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=24906146)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4343198)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Development%20and%20applications%20of%20CRISPR-Cas9%20for%20genome%20engineering&journal=Cell&volume=157&pages=1262-1278&publication_year=2014&author=Hsu%2CPD&author=Lander%2CES&author=Zhang%2CF)
23. Sander, J. D. & Joung, J. K. CRISPR-Cas systems for editing, regulating and targeting genomes. *Nat. Biotechnol.* **32**, 347–355 (2014).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC2cXjtlyrsLo%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=24584096)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4022601)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPR-Cas%20systems%20for%20editing%2C%20regulating%20and%20targeting%20genomes&journal=Nat.%20Biotechnol.&volume=32&pages=347-355&publication_year=2014&author=Sander%2CJD&author=Joung%2CJK)
24. Shalem, O., Sanjana, N. E. & Zhang, F. High-throughput functional genomics using CRISPR–Cas9. *Nat. Rev. Genet.* **16**, 299–311 (2015).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC2MXmt1Smtrg%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=25854182)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4503232)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=High-throughput%20functional%20genomics%20using%20CRISPR%E2%80%93Cas9&journal=Nat.%20Rev.%20Genet.&volume=16&pages=299-311&publication_year=2015&author=Shalem%2CO&author=Sanjana%2CNE&author=Zhang%2CF)
25. Qi, L. S. et al. Repurposing CRISPR as an RNA-guided platform for sequence-specific control of gene expression. *Cell* **152**, 1173–1183 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXjsFChs7s%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23452860)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3664290)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Repurposing%20CRISPR%20as%20an%20RNA-guided%20platform%20for%20sequence-specific%20control%20of%20gene%20expression&journal=Cell&volume=152&pages=1173-1183&publication_year=2013&author=Qi%2CLS)
26. Gilbert, L. A. et al. CRISPR-mediated modular RNA-guided regulation of transcription in eukaryotes. *Cell* **154**, 442–451 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXhtV2ru7fP)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23849981)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3770145)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPR-mediated%20modular%20RNA-guided%20regulation%20of%20transcription%20in%20eukaryotes&journal=Cell&volume=154&pages=442-451&publication_year=2013&author=Gilbert%2CLA)
27. Konermann, S. et al. Optical control of mammalian endogenous transcription and epigenetic states. *Nature* **500**, 472–476 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXhtlWjt7rO)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23877069)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3856241)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Optical%20control%20of%20mammalian%20endogenous%20transcription%20and%20epigenetic%20states&journal=Nature&volume=500&pages=472-476&publication_year=2013&author=Konermann%2CS)
28. Maeder, M. L. et al. CRISPR RNA-guided activation of endogenous human genes. *Nat. Methods* **10**, 977–979 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXhtFOhs7bI)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23892898)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3794058)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPR%20RNA-guided%20activation%20of%20endogenous%20human%20genes&journal=Nat.%20Methods&volume=10&pages=977-979&publication_year=2013&author=Maeder%2CML)
29. Perez-Pinera, P. et al. RNA-guided gene activation by CRISPR-Cas9-based transcription factors. *Nat. Methods* **10**, 973–976 (2013).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC3sXhtFOhtLnL)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=23892895)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3911785)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=RNA-guided%20gene%20activation%20by%20CRISPR-Cas9-based%20transcription%20factors&journal=Nat.%20Methods&volume=10&pages=973-976&publication_year=2013&author=Perez-Pinera%2CP)
30. Anzalone, A. V. et al. Search-and-replace genome editing without double-strand breaks or donor DNA. *Nature* **576**, 149–157 (2019).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1MXitFGns7rO)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=31634902)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6907074)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Search-and-replace%20genome%20editing%20without%20double-strand%20breaks%20or%20donor%20DNA&journal=Nature&volume=576&pages=149-157&publication_year=2019&author=Anzalone%2CAV)
31. Gaudelli, N. M. et al. Programmable base editing of A•T to G•C in genomic DNA without DNA cleavage. *Nature* **551**, 464–471 (2017).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC2sXhvVKqtb3P)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=29160308)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5726555)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Programmable%20base%20editing%20of%20A%E2%80%A2T%20to%20G%E2%80%A2C%20in%20genomic%20DNA%20without%20DNA%20cleavage&journal=Nature&volume=551&pages=464-471&publication_year=2017&author=Gaudelli%2CNM)
32. Anzalone, A. V., Koblan, L. W. & Liu, D. R. Genome editing with CRISPR–Cas nucleases, base editors, transposases and prime editors. *Nat. Biotechnol.* **38**, 824–844 (2020).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXht1WrtLvK)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=32572269)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Genome%20editing%20with%20CRISPR%E2%80%93Cas%20nucleases%2C%20base%20editors%2C%20transposases%20and%20prime%20editors&journal=Nat.%20Biotechnol.&volume=38&pages=824-844&publication_year=2020&author=Anzalone%2CAV&author=Koblan%2CLW&author=Liu%2CDR)
33. Pickar-Oliver, A. & Gersbach, C. A. The next generation of CRISPR–Cas technologies and applications. *Nat. Rev. Mol. Cell Biol.* **20**, 490–507 (2019).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1MXhtVOju7jM)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=31147612)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC7079207)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=The%20next%20generation%20of%20CRISPR%E2%80%93Cas%20technologies%20and%20applications&journal=Nat.%20Rev.%20Mol.%20Cell%20Biol.&volume=20&pages=490-507&publication_year=2019&author=Pickar-Oliver%2CA&author=Gersbach%2CCA)
34. Tsang, J. 15 Years of Addgene: The Top 15 Plasmids. *Addgene Blog* <https://blog.addgene.org/15-years-of-addgene-the-top-15-plasmids> (2019).
35. Office of the Commissioner. FDA approves first gene therapies to treat patients with sickle cell disease. *FDA News Release* <https://www.fda.gov/news-events/press-announcements/fda-approves-first-gene-therapies-treat-patients-sickle-cell-disease> (2024).
36. Office of the Commissioner. FDA roundup: January 16, 2024. *FDA News Release* <https://www.fda.gov/news-events/press-announcements/fda-roundup-january-16-2024> (2024).
37. Wei, J. et al. Chain-of-thought prompting elicits reasoning in large language models. In *Advances in Neural Information Processing Systems* (eds Koyejo, S. et al.) Vol. 35, 24824–24837 (Curran Associates, Inc., 2022).
38. Zhou, D. et al. Least-to-most prompting enables complex reasoning in large language models. In *Proc. 11th International Conference on Learning Representations* (OpenReview, 2023).
39. Kim, H. K. et al. Deep learning improves prediction of CRISPR–Cpf1 guide RNA activity. *Nat. Biotechnol.* **36**, 239–241 (2018).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1cXhvVequrw%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=29431740)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Deep%20learning%20improves%20prediction%20of%20CRISPR%E2%80%93Cpf1%20guide%20RNA%20activity&journal=Nat.%20Biotechnol.&volume=36&pages=239-241&publication_year=2018&author=Kim%2CHK)
40. DeWeirdt, P. C. et al. Optimization of AsCas12a for combinatorial genetic screens in human cells. *Nat. Biotechnol.* **39**, 94–104 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXhtlyntbrF)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=32661438)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Optimization%20of%20AsCas12a%20for%20combinatorial%20genetic%20screens%20in%20human%20cells&journal=Nat.%20Biotechnol.&volume=39&pages=94-104&publication_year=2021&author=DeWeirdt%2CPC)
41. Doench, J. G. et al. Optimized sgRNA design to maximize activity and minimize off-target effects of CRISPR-Cas9. *Nat. Biotechnol.* **34**, 184–191 (2016).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC28Xps1GisQ%3D%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=26780180)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4744125)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Optimized%20sgRNA%20design%20to%20maximize%20activity%20and%20minimize%20off-target%20effects%20of%20CRISPR-Cas9&journal=Nat.%20Biotechnol.&volume=34&pages=184-191&publication_year=2016&author=Doench%2CJG)
42. Sanson, K. R. et al. Optimized libraries for CRISPR-Cas9 genetic screens with multiple modalities. *Nat. Commun.* **9**, 5416 (2018).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1cXisFyntLjM)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=30575746)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6303322)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Optimized%20libraries%20for%20CRISPR-Cas9%20genetic%20screens%20with%20multiple%20modalities&journal=Nat.%20Commun.&volume=9&publication_year=2018&author=Sanson%2CKR)
43. Labun, K. et al. CHOPCHOP v3: expanding the CRISPR web toolbox beyond genome editing. *Nucleic Acids Res.* **47**, W171–W174 (2019).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXktVyiu7c%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=31106371)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6602426)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CHOPCHOP%20v3%3A%20expanding%20the%20CRISPR%20web%20toolbox%20beyond%20genome%20editing&journal=Nucleic%20Acids%20Res.&volume=47&pages=W171-W174&publication_year=2019&author=Labun%2CK)
44. Wang, Z. et al. Chain-of-table: evolving tables in the reasoning chain for table understanding. In *Proc. 12th International Conference on Learning Representations* (OpenReview, 2024).
45. Shi, J. et al. Discovery of cancer drug targets by CRISPR-Cas9 screening of protein domains. *Nat. Biotechnol.* **33**, 661–667 (2015).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC2MXhtFemtr/J)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=25961408)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4529991)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Discovery%20of%20cancer%20drug%20targets%20by%20CRISPR-Cas9%20screening%20of%20protein%20domains&journal=Nat.%20Biotechnol.&volume=33&pages=661-667&publication_year=2015&author=Shi%2CJ)
46. Giuliano, C. J., Lin, A., Girish, V. & Sheltzer, J. M. Generating single cell-derived knockout clones in mammalian cells with CRISPR/Cas9. *Curr. Protoc. Mol. Biol.* **128**, e100 (2019).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=31503414)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6741428)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Generating%20single%20cell-derived%20knockout%20clones%20in%20mammalian%20cells%20with%20CRISPR%2FCas9&journal=Curr.%20Protoc.%20Mol.%20Biol.&volume=128&publication_year=2019&author=Giuliano%2CCJ&author=Lin%2CA&author=Girish%2CV&author=Sheltzer%2CJM)
47. Du, D. & Qi, L. S. CRISPR technology for genome activation and repression in mammalian cells. *Cold Spring Harb. Protoc.* **2016**, pdb.prot090175 (2016).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=26729910)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPR%20technology%20for%20genome%20activation%20and%20repression%20in%20mammalian%20cells&journal=Cold%20Spring%20Harb.%20Protoc.&volume=2016&publication_year=2016&author=Du%2CD&author=Qi%2CLS)
48. Huang, T. P., Newby, G. A. & Liu, D. R. Precision genome editing using cytosine and adenine base editors in mammalian cells. *Nat. Protoc.* **16**, 1089–1128 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3MXhsFOju7c%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=33462442)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Precision%20genome%20editing%20using%20cytosine%20and%20adenine%20base%20editors%20in%20mammalian%20cells&journal=Nat.%20Protoc.&volume=16&pages=1089-1128&publication_year=2021&author=Huang%2CTP&author=Newby%2CGA&author=Liu%2CDR)
49. Hanna, R. E. et al. Massively parallel assessment of human variants with base editor screens. *Cell* **184**, 1064–1080.e20 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3MXks1Kltr0%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=33606977)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Massively%20parallel%20assessment%20of%20human%20variants%20with%20base%20editor%20screens&journal=Cell&volume=184&pages=1064-1080.e20&publication_year=2021&author=Hanna%2CRE)
50. Doman, J. L., Sousa, A. A., Randolph, P. B., Chen, P. J. & Liu, D. R. Designing and executing prime editing experiments in mammalian cells. *Nat. Protoc.* **17**, 2431–2468 (2022).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB38XitV2lt73J)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=35941224)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC9799714)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Designing%20and%20executing%20prime%20editing%20experiments%20in%20mammalian%20cells&journal=Nat.%20Protoc.&volume=17&pages=2431-2468&publication_year=2022&author=Doman%2CJL&author=Sousa%2CAA&author=Randolph%2CPB&author=Chen%2CPJ&author=Liu%2CDR)
51. McGee, A. V. et al. Modular vector assembly enables rapid assessment of emerging CRISPR technologies. *Cell Genomics* **4**, 100519 (2024).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB2cXlvVemtLw%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=38484704)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC10943585)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Modular%20vector%20assembly%20enables%20rapid%20assessment%20of%20emerging%20CRISPR%20technologies&journal=Cell%20Genomics&volume=4&publication_year=2024&author=McGee%2CAV)
52. Untergasser, A. et al. Primer3—new capabilities and interfaces. *Nucleic Acids Res.* **40**, e115 (2012).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC38Xht1Kjs7nF)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=22730293)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3424584)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Primer3%E2%80%94new%20capabilities%20and%20interfaces&journal=Nucleic%20Acids%20Res.&volume=40&publication_year=2012&author=Untergasser%2CA)
53. Cancellieri, S., Canver, M. C., Bombieri, N., Giugno, R. & Pinello, L. CRISPRitz: rapid, high-throughput and variant-aware in silico off-target site identification for CRISPR genome editing. *Bioinformatics* **36**, 2001–2008 (2020).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXitlOlsrzE)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=31764961)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPRitz%3A%20rapid%2C%20high-throughput%20and%20variant-aware%20in%20silico%20off-target%20site%20identification%20for%20CRISPR%20genome%20editing&journal=Bioinformatics&volume=36&pages=2001-2008&publication_year=2020&author=Cancellieri%2CS&author=Canver%2CMC&author=Bombieri%2CN&author=Giugno%2CR&author=Pinello%2CL)
54. Hwang, G.-H. et al. PE-Designer and PE-Analyzer: web-based design and analysis tools for CRISPR prime editing. *Nucleic Acids Res.* **49**, W499–W504 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3MXhvV2isLfL)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=33939828)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC8265102)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=PE-Designer%20and%20PE-Analyzer%3A%20web-based%20design%20and%20analysis%20tools%20for%20CRISPR%20prime%20editing&journal=Nucleic%20Acids%20Res.&volume=49&pages=W499-W504&publication_year=2021&author=Hwang%2CG-H)
55. Chow, R. D., Chen, J. S., Shen, J. & Chen, S. A web tool for the design of prime-editing guide RNAs. *Nat. Biomed. Eng.* **5**, 190–194 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXhvFyisrbJ)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=32989284)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=A%20web%20tool%20for%20the%20design%20of%20prime-editing%20guide%20RNAs&journal=Nat.%20Biomed.%20Eng.&volume=5&pages=190-194&publication_year=2021&author=Chow%2CRD&author=Chen%2CJS&author=Shen%2CJ&author=Chen%2CS)
56. Morris, J. A., Rahman, J. A., Guo, X. & Sanjana, N. E. Automated design of CRISPR prime editors for 56,000 human pathogenic variants. *iScience* **24**, 103380 (2021).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB38Xmt1Wisrc%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=35814872)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC9259959)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Automated%20design%20of%20CRISPR%20prime%20editors%20for%2056%2C000%20human%20pathogenic%20variants&journal=iScience&volume=24&publication_year=2021&author=Morris%2CJA&author=Rahman%2CJA&author=Guo%2CX&author=Sanjana%2CNE)
57. Clement, K. et al. CRISPResso2 provides accurate and rapid genome editing sequence analysis. *Nat. Biotechnol.* **37**, 224–226 (2019).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1MXosV2rtb0%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=30809026)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6533916)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=CRISPResso2%20provides%20accurate%20and%20rapid%20genome%20editing%20sequence%20analysis&journal=Nat.%20Biotechnol.&volume=37&pages=224-226&publication_year=2019&author=Clement%2CK)
58. Introducing Meta Llama 3: the most capable openly available LLM to date. *Meta Blog* <https://ai.meta.com/blog/meta-llama-3/> (2024).
59. Vaswani, A. et al. Attention is all you need. *Adv. Neural Inf. Process. Syst.* **30**, 5998–6008 (2017).
60. Loshchilov, I. et al. *Decoupled Weight Decay Regularization* (ICLR, 2019).
61. Hu, E. et al. *LoRA: Low-Rank Adaptation of Large Language Models* (ICLR, 2022).
62. Dettmers, T., Pagnoni, A., Holtzman, A. & Zettlemoyer, L. QLORA: efficient finetuning of quantized LLMs. In *Proc. 37th International Conference on Neural Information Processing Systems* (eds Oh, A. et al.) 441 (NeurIPS, 2023).
63. Zhou, C. et al. Lima: Less is more for alignment. In *Proc. 37th International Conference on Neural Information Processing Systems* (eds Oh, A. et al.) 2400 (NeurIPS, 2023).
64. Delvin, J., Chang, M.-W., Lee, K. & Toutanova, K. BERT: pre-training of deep bidirectional transformers for language understanding. *Proc. Conf. NAACL-HLT* **2019**, 4171–4186 (2019).
65. Liu, Z., Lin, W., Shi, Y. & Zhao, J. RoBERTa: A Robustly Optimized BERT Pretraining Approach. In *Proc. 20th Chinese National Conference on Computational Linguistics* (eds Li, S. et al.) 1218–1227 (Chinese Information Processing Society of China, 2021).
66. *Frequently Asked Questions About CRISPR* (Innovative Genomics Institute, 2025).
67. Beumer, J. et al. CRISPR: questions and answers. *STAR Protoc.* <https://star-protocols.cell.com/protocols/2555> (2023).
68. Doudna, J. A. The promise and challenge of therapeutic genome editing. *Nature* **578**, 229–236 (2020).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3cXktlCrt7o%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=32051598)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC8992613)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=The%20promise%20and%20challenge%20of%20therapeutic%20genome%20editing&journal=Nature&volume=578&pages=229-236&publication_year=2020&author=Doudna%2CJA)
69. Hoffmann, S. A. et al. Safety by design: biosafety and biosecurity in the age of synthetic genomics. *iScience* **26**, 106165 (2023).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BB3sXktVyqsL0%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=36895643)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC9988571)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Safety%20by%20design%3A%20biosafety%20and%20biosecurity%20in%20the%20age%20of%20synthetic%20genomics&journal=iScience&volume=26&publication_year=2023&author=Hoffmann%2CSA)
70. Lander, E. S. et al. Adopt a moratorium on heritable genome editing. *Nature* **567**, 165–168 (2019).

    [CAS](/articles/cas-redirect/1%3ACAS%3A528%3ADC%2BC1MXnvF2hsLg%3D)
    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=30867611)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Adopt%20a%20moratorium%20on%20heritable%20genome%20editing&journal=Nature&volume=567&pages=165-168&publication_year=2019&author=Lander%2CES)
71. *Summary of the HIPAA Privacy Rule* <https://www.hhs.gov/hipaa/for-professionals/privacy/laws-regulations/index.html> (US Department of Health and Human Services, 2008).
72. Naveed, M. et al. Privacy in the genomic era. *ACM Comput. Surv.* **48**, 6 (2015).

    [PubMed](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=26640318)
    [PubMed Central](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4666540)
    [Google Scholar](http://scholar.google.com/scholar_lookup?&title=Privacy%20in%20the%20genomic%20era&journal=ACM%20Comput.%20Surv.&volume=48&publication_year=2015&author=Naveed%2CM)
73. OpenAI. ChatGPT plugins. <https://openai.com/blog/chatgpt-plugins> (2023).
74. Shen, Y. et al. HuggingGPT: solving AI tasks with ChatGPT and its friends in hugging face. In *Advances in Neural Information Processing Systems* (eds Oh, A. et al.) Vol. 36, 38154–38180 (Curran Associates, Inc., 2023).
75. Wang, G. et al. Voyager: an open-ended embodied agent with large language models. Preprint at <https://doi.org/10.48550/arXiv.2305.16291> (2023).
76. Hao, S., Liu, T., Wang, Z. & Hu, Z. ToolkenGPT: augmenting frozen language models with massive tools via tool embeddings. In *Advances in Neural Information Processing Systems* (eds Oh, A. et al.) Vol. 36, 45870–45894 (Curran Associates, Inc., 2023).
77. Patil, S. G., Zhang, T., Wang, X. & Gonzalez, J. E. Gorilla: large language model connected with massive APIs. In *Advances in Neural Information Processing Systems* (eds Globerson, A. et al.) Vol. 37, 126544–126565 (Curran Associates, Inc., 2024).
78. Qu, Y. et al. CRISPR-GPT: an LLM agent for automated design of gene-editing experiments. *GitHub* <https://github.com/cong-lab/crispr-gpt-pub> (2024).

[Download references](https://citation-needed.springer.com/v2/references/10.1038/s41551-025-01463-z?format=refman&flavour=references)

## Acknowledgements

This work was supported by the National Institutes of Health (grant nos. 1R35HG011316 and 1R01GM141627 to L.C.), Donald and Delia Baxter Foundation Faculty Scholar award (L.C.), the Weintz family foundation (L.C.), and National Science Foundation grant 1653435 (M.W.). This work used supercomputing resources provided by Nebius Group N.V. and the Stanford Genetics Bioinformatics Service Center, supported by NIH Instrumentation Grant S10 OD023452. We thank the Stanford Institute for Stem Cell Biology and Regenerative Medicine, the Stanford Cancer Institute, and the Stanford Wu-Tsai Neuroscience Institute for technical advice and support.

## Author information

Author notes

1. These authors contributed equally: Yuanhao Qu, Kaixuan Huang.

### Authors and Affiliations

1. Department of Pathology, Department of Genetics, Cancer Biology Program, Stanford University School of Medicine, Stanford, CA, USA

   Yuanhao Qu, Di Yin, William A. Johnson, Xiaotong Wang & Le Cong
2. Center for Statistics and Machine Learning, Department of Electrical and Computer Engineering, Princeton University, Princeton, NJ, USA

   Kaixuan Huang, Ming Yin & Mengdi Wang
3. Department of Computing, Data Science, and Society, University of California, Berkeley, Berkeley, CA, USA

   Kanghong Zhan
4. Department of Computer Science, University of California, Berkeley, Berkeley, CA, USA

   Dyllan Liu & Russ B. Altman
5. Department of Medicine, Stanford University School of Medicine, Stanford, CA, USA

   Henry C. Cousins & Mihir Shah
6. Medical Scientist Training Program, Stanford University School of Medicine, Stanford, CA, USA

   Henry C. Cousins
7. Department of Bioengineering, Department of Genetics, Stanford University, Stanford, CA, USA

   Russ B. Altman
8. Google DeepMind, Mountain View, CA, USA

   Denny Zhou

Authors

1. Yuanhao Qu[View author publications](/search?author=Yuanhao%20Qu)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Yuanhao%20Qu) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Yuanhao%20Qu%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
2. Kaixuan Huang[View author publications](/search?author=Kaixuan%20Huang)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Kaixuan%20Huang) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Kaixuan%20Huang%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
3. Ming Yin[View author publications](/search?author=Ming%20Yin)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Ming%20Yin) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Ming%20Yin%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
4. Kanghong Zhan[View author publications](/search?author=Kanghong%20Zhan)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Kanghong%20Zhan) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Kanghong%20Zhan%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
5. Dyllan Liu[View author publications](/search?author=Dyllan%20Liu)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Dyllan%20Liu) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Dyllan%20Liu%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
6. Di Yin[View author publications](/search?author=Di%20Yin)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Di%20Yin) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Di%20Yin%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
7. Henry C. Cousins[View author publications](/search?author=Henry%20C.%20Cousins)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Henry%20C.%20Cousins) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Henry%20C.%20Cousins%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
8. William A. Johnson[View author publications](/search?author=William%20A.%20Johnson)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=William%20A.%20Johnson) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22William%20A.%20Johnson%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
9. Xiaotong Wang[View author publications](/search?author=Xiaotong%20Wang)

   Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Xiaotong%20Wang) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Xiaotong%20Wang%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
10. Mihir Shah[View author publications](/search?author=Mihir%20Shah)

    Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Mihir%20Shah) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Mihir%20Shah%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
11. Russ B. Altman[View author publications](/search?author=Russ%20B.%20Altman)

    Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Russ%20B.%20Altman) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Russ%20B.%20Altman%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
12. Denny Zhou[View author publications](/search?author=Denny%20Zhou)

    Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Denny%20Zhou) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Denny%20Zhou%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
13. Mengdi Wang[View author publications](/search?author=Mengdi%20Wang)

    Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Mengdi%20Wang) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Mengdi%20Wang%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
14. Le Cong[View author publications](/search?author=Le%20Cong)

    Search author on:[PubMed](https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&term=Le%20Cong) [Google Scholar](https://scholar.google.co.uk/scholar?as_q=&num=10&btnG=Search+Scholar&as_epq=&as_oq=&as_eq=&as_occt=any&as_sauthors=%22Le%20Cong%22&as_publication=&as_ylo=&as_yhi=&as_allsubj=all&hl=en)
### Contributions

Y.Q., K.H., M.Y., M.W. and L.C. designed the research, constructed models, performed experiments and analysed data. K.Z., D.L. and D.Z. supported model construction, deployment and evaluation. Y.Q., X.W., D.Y, M.Y., H.C.C., W.A.J., M.S. and R.B.A. contributed to human evaluation and analysed data. L.C. and M.W. conceived and supervised the research. Y.Q., K.H., M.Y., M.W. and L.C. wrote the paper with input from all authors.

### Corresponding authors

Correspondence to
Mengdi Wang or Le Cong.

## Ethics declarations

### Competing interests

Princeton University and Stanford University have filed patent applications (#19/093,928, Princeton and Stanford, 2025) based on this work, where L.C., M.W., Y.Q. and K.H. are listed as inventors. L.C. is a member of the scientific advisory board of Arbor Biotechnologies. L.C. has equity interest in Auto Bio, Rootpath Genomics, and Acrobat Genomics. D.Z. is an employee of Google DeepMind. The remaining authors declare no competing interests.

## Peer review

### Peer review information

*Nature Biomedical Engineering* thanks the anonymous reviewers for their contribution to the peer review of this work.

## Additional information

**Publisher’s note** Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.

## Extended data

### [Extended Data Fig. 1 LLM-powered exon suggestion for guideRNA design in CRISPR-GPT.](/articles/s41551-025-01463-z/figures/7)

**a**, Workflow of the exon suggestion feature within the guideRNA design module in CRISPR-GPT. **b**. Demonstration of the thought and action processes for exon suggestion in response to a real-world user request (additional details in Methods) **c**. Comparison of gRNA designs with exon information from different tools (CRISPR-GPT, CRISPick, CHOPCHOP). Notes on the designs are provided. CRISPick designs used the default combined ranking by on-/off-target scores based on Azimuth 2.0 algorithm and Cutting Frequency Determination (CFD) scores. CHOPCHOP design uses the default on-target efficiency ranking based on Doench et al., 2016. **d**, Exons and key bromodomains (BD1/BD2) of the BRD4 protein are shown for reference.

### [Extended Data Fig. 2 Example of CRISPR-GPT QA on gene-editing questions, compared to baseline models (Part 1/3).](/articles/s41551-025-01463-z/figures/8)

The question was from the gene-editing QA test set (Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). Questions and answers are displayed, with expert notes in bottom boxes. Evaluations and comments conducted in a fully blinded setting.

### [Extended Data Fig. 3 Example of CRISPR-GPT QA on gene-editing questions, compared to baseline models (Part 2/3).](/articles/s41551-025-01463-z/figures/9)

The question was from the gene-editing QA test set (Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). Questions and answers are displayed, with expert notes in bottom boxes. Evaluations and comments conducted in a fully blinded setting.

### [Extended Data Fig. 4 Example of CRISPR-GPT QA on gene-editing questions, compared to baseline models (Part 3/3).](/articles/s41551-025-01463-z/figures/10)

The question was from the gene-editing QA test set (Supplementary Note [C](/articles/s41551-025-01463-z#MOESM1)). Questions and answers are displayed, with expert notes in bottom boxes. Evaluations and comments conducted in a fully blinded setting.

### [Extended Data Fig. 5 Examples of CRISPR-GPT defending against dual usage and ethical, privacy risks.](/articles/s41551-025-01463-z/figures/11)

The agent identifies potential risks related to dual usage risks (**top**), human heritable gene-editing and private genetic information leakage (**bottom**), responds with warning messages or errors, and stops proceeding.

## Supplementary information

### [Supplementary Information](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM1_ESM.pdf)

Supplementary Tables 1–4, Figs. 1–5 and Notes A–F.

### [Reporting Summary](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM2_ESM.pdf)

### [Supplementary Data 1](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM3_ESM.zip)

AI agent demos as chat history.

### [Supplementary Data 2](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM4_ESM.xlsx)

DURC information for AI agent safety.

### [Supplementary Video 1](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM5_ESM.mp4)

AI agent demos as recorded video.

### [Supplementary Video 2](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM6_ESM.mp4)

AI agent demos as recorded video.

### [Supplementary Video 3](https://static-content.springer.com/esm/art%3A10.1038/s41551-025-01463-z/MediaObjects/41551_2025_1463_MOESM7_ESM.mov)

AI agent demos as recorded video.

## Rights and permissions

**Open Access** This article is licensed under a Creative Commons Attribution 4.0 International License, which permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or other third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line to the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this licence, visit <http://creativecommons.org/licenses/by/4.0/>.

[Reprints and permissions](https://s100.copyright.com/AppDispatchServlet?title=CRISPR-GPT%20for%20agentic%20automation%20of%20gene-editing%20experiments&author=Yuanhao%20Qu%20et%20al&contentID=10.1038%2Fs41551-025-01463-z&copyright=The%20Author%28s%29&publication=2157-846X&publicationDate=2025-07-30&publisherName=SpringerNature&orderBeanReset=true&oa=CC%20BY)

## About this article

[![Check for updates. Verify currency and authenticity via CrossMark](data:image/svg+xml;base64...)](https://crossmark.crossref.org/dialog/?doi=10.1038/s41551-025-01463-z)
### Cite this article

Qu, Y., Huang, K., Yin, M. *et al.* CRISPR-GPT for agentic automation of gene-editing experiments.
*Nat. Biomed. Eng* (2025). https://doi.org/10.1038/s41551-025-01463-z

[Download citation](https://citation-needed.springer.com/v2/references/10.1038/s41551-025-01463-z?format=refman&flavour=citation)

* Received: 25 June 2024
* Accepted: 17 June 2025
* Published: 30 July 2025
* DOI: https://doi.org/10.1038/s41551-025-01463-z

### Share this article

Anyone you share the following link with will be able to read this content:

Get shareable link

Sorry, a shareable link is not currently available for this article.

Copy to clipboard

Provided by the Springer Nature SharedIt content-sharing initiative

[Download PDF](/articles/s41551-025-01463-z.pdf)

Advertisement

[![Advertisement](//pubads.g.doubleclick.net/gampad/ad?iu=/285/natbiomedeng.nature.com/article&sz=300x250&c=-1444146478&t=pos%3Dright%26type%3Darticle%26artid%3Ds41551-025-01463-z%26doi%3D10.1038/s41551-025-01463-z%26techmeta%3D1,109,13,22,23,31,38,39,41,42,44,45,56,90%26subjmeta%3D1042,114,208,337,4041,631,639,705%26kwrd%3DComputational+biology+and+bioinformatics,Computational+science,CRISPR-Cas+systems)](//pubads.g.doubleclick.net/gampad/jump?iu=/285/natbiomedeng.nature.com/article&sz=300x250&c=-1444146478&t=pos%3Dright%26type%3Darticle%26artid%3Ds41551-025-01463-z%26doi%3D10.1038/s41551-025-01463-z%26techmeta%3D1,109,13,22,23,31,38,39,41,42,44,45,56,90%26subjmeta%3D1042,114,208,337,4041,631,639,705%26kwrd%3DComputational+biology+and+bioinformatics,Computational+science,CRISPR-Cas+systems)

## Explore content

* [Research articles](/natbiomedeng/research-articles)
* [Reviews & Analysis](/natbiomedeng/reviews-and-analysis)
* [News & Comment](/natbiomedeng/news-and-comment)
* [Videos](/natbiomedeng/videos)
* [Current issue](/natbiomedeng/current-issue)
* [Collections](/natbiomedeng/collections)

* [Follow us on Facebook](https://www.facebook.com/natBME)
* [Follow us on Twitter](https://twitter.com/natBME)
* [Sign up for alerts](https://www.nature.com/my-account/alerts/subscribe-journal?list-id=377)
* [RSS feed](https://www.nature.com/natbiomedeng.rss)

## About the journal

* [Aims & Scope](/natbiomedeng/aims)
* [Journal Information](/natbiomedeng/journal-information)
* [Journal Metrics](/natbiomedeng/journal-impact)
* [About the Editors](/natbiomedeng/editors)
* [Research Cross-Journal Editorial Team](/natbiomedeng/research-cross-journal-editorial-team)
* [Reviews Cross-Journal Editorial Team](/natbiomedeng/reviews-cross-journal-editorial-team)
* [Our publishing models](/natbiomedeng/our-publishing-models)
* [Editorial Values Statement](/natbiomedeng/editorial-values-statement)
* [Editorial Policies](/natbiomedeng/editorial-policies)
* [Content Types](/natbiomedeng/content)
* [Contact](/natbiomedeng/contact)

## Publish with us

* [Submission Guidelines](/natbiomedeng/submission-guidelines)
* [For Reviewers](/natbiomedeng/for-reviewers)
* [Language editing services](https://authorservices.springernature.com/go/sn/?utm_source=For+Authors&utm_medium=Website_Nature&utm_campaign=Platform+Experimentation+2022&utm_id=PE2022)
* [Open access funding](/natbiomedeng/open-access-funding)
* [Submit manuscript](https://mts-natbiomedeng.nature.com)

## Search

Search articles by subject, keyword or author

Show results from
All journals
This journal

Search

[Advanced search](/search/advanced)
### Quick links

* [Explore articles by subject](/subjects)
* [Find a job](/naturecareers)
* [Guide to authors](/authors/index.html)
* [Editorial policies](/authors/editorial_policies/)

Nature Biomedical Engineering

(*Nat. Biomed. Eng*)

ISSN 2157-846X (online)

## nature.com sitemap

### About Nature Portfolio

* [About us](https://www.nature.com/npg_/company_info/index.html)
* [Press releases](https://www.nature.com/npg_/press_room/press_releases.html)
* [Press office](https://press.nature.com/)
* [Contact us](https://support.nature.com/support/home)

### Discover content

* [Journals A-Z](https://www.nature.com/siteindex)
* [Articles by subject](https://www.nature.com/subjects)
* [protocols.io](https://www.protocols.io/)
* [Nature Index](https://www.natureindex.com/)

### Publishing policies

* [Nature portfolio policies](https://www.nature.com/authors/editorial_policies)
* [Open access](https://www.nature.com/nature-research/open-access)

### Author & Researcher services

* [Reprints & permissions](https://www.nature.com/reprints)
* [Research data](https://www.springernature.com/gp/authors/research-data)
* [Language editing](https://authorservices.springernature.com/language-editing/)
* [Scientific editing](https://authorservices.springernature.com/scientific-editing/)
* [Nature Masterclasses](https://masterclasses.nature.com/)
* [Research Solutions](https://solutions.springernature.com/)

### Libraries & institutions

* [Librarian service & tools](https://www.springernature.com/gp/librarians/tools-services)
* [Librarian portal](https://www.springernature.com/gp/librarians/manage-your-account/librarianportal)
* [Open research](https://www.nature.com/openresearch/about-open-access/information-for-institutions)
* [Recommend to library](https://www.springernature.com/gp/librarians/recommend-to-your-library)

### Advertising & partnerships

* [Advertising](https://partnerships.nature.com/product/digital-advertising/)
* [Partnerships & Services](https://partnerships.nature.com/)
* [Media kits](https://partnerships.nature.com/media-kits/)
* [Branded
  content](https://partnerships.nature.com/product/branded-content-native-advertising/)

### Professional development

* [Nature Awards](https://www.nature.com/immersive/natureawards/index.html)
* [Nature Careers](https://www.nature.com/naturecareers/)
* [Nature
  Conferences](https://conferences.nature.com)

### Regional websites

* [Nature Africa](https://www.nature.com/natafrica)
* [Nature China](http://www.naturechina.com)
* [Nature India](https://www.nature.com/nindia)
* [Nature Japan](https://www.natureasia.com/ja-jp)
* [Nature Middle East](https://www.nature.com/nmiddleeast)

* [Privacy
  Policy](https://www.nature.com/info/privacy)
* [Use
  of cookies](https://www.nature.com/info/cookies)
* Your privacy choices/Manage cookies
* [Legal
  notice](https://www.nature.com/info/legal-notice)
* [Accessibility
  statement](https://www.nature.com/info/accessibility-statement)
* [Terms & Conditions](https://www.nature.com/info/terms-and-conditions)
* [Your US state privacy rights](https://www.springernature.com/ccpa)

[![Springer Nature](/static/images/logos/sn-logo-white-ea63208b81.svg)](https://www.springernature.com/)

© 2025 Springer Nature Limited

xml version="1.0" encoding="UTF-8"?

Close banner

Close

![Nature Briefing Microbiology](/static/images/logos/nature-briefing-microbiology-logo-3fa570e5fb.svg)

Sign up for the *Nature Briefing: Microbiology* newsletter — what matters in microbiology research, free to your inbox weekly.

Email address

Sign up

I agree my information will be processed in accordance with the *Nature* and Springer Nature Limited [Privacy Policy](https://www.nature.com/info/privacy).

Close banner

Close

Get the most important science stories of the day, free in your inbox.
[Sign up for Nature Briefing: Microbiology](/briefing/microbiology/?brieferEntryPoint=MicrobiologyBriefingBanner)

![](https://verify.nature.com/verify/nature.png)